Title: Aligning & Mapping RNAseq Reads - Dr. Rob Patro
Date: 2016-08-15 11:31
Author: monsterbashseq
Category: Uncategorized
Slug: 4089
Status: draft

http://robpatro.com/redesign/AlignmentAndMapping.pdf

RNAseq, many uses

Explains edit distance, between 2 sequences, e.g. riddle -&gt; ridle
-&gt; riple -&gt; triple

Big O notation: based on read size, length and bases

O(N\*l\*L) where N = reads, l = length, L = bases

razor3, massai = optimal alignments

bowtie is heuristic = work well, not guaranteed

Software run algorithms because we can't know exactly the problem.

Phylogeny of Read-Alignment, awesome slide !!!!!

![alignment\_phylogeny](https://monsterbashseq.files.wordpress.com/2016/08/alignment_phylogeny.png){.alignnone
.size-full .wp-image-4106 width="1125" height="807"}

Recommends STAR, fast and good algorithms.

TopHat2 deprecated in favor of HISAT, implemented in STAR

paper Pertea and Mihaela et al 2016. Nature Protocols 11, 1650-1667

Mapping, where and how read corresponds to reference.

RapMap, generalized suffix array similar to STAR to build up lookup
table of kmers to speed things up

What's the difference between quasi-mapping and pseudoalignment?

-   pseudoalignment (Kallisto), given read just want transcripts that
    are compatible. Can compute what they are without where.Kmer data
    structure
-   quasimapping: transcript position, orientation. have to have
    positions before transcripts, suffix array data structure

recall, precision F1

RapClust?

Tutorial:

http://angus.readthedocs.io/en/2016/rob\_map/tut.html

 

 

 
