Title: Error with HTSeq
Date: 2014-07-15 10:10
Author: monsterbashseq
Category: Bioinformatics, software
Slug: 869
Status: published

Running HTSeq to get raw read counts associated with chromosomal
positions/annotated genes from sorted .bam files:

    Error occurred when processing SAM input (record #30675291 in file {PATH}/tophat/Exp-1-Basal/accepted_hits.sorted.bam):
     Maximum alignment buffer size exceeded while pairing SAM alignments.
     [Exception type: ValueError, raised in __init__.py:671]

Simon Anders, the author of the package from EMBL Heidelberg, suggested
in
this [forum post](http://seqanswers.com/forums/showthread.php?t=41531) to
remove the "-r pos" or "--order=pos" argument. I did this and it seems
to work fine now. Interesting that I had run HTSeq before with a
different data set of sorted bam files and did not get this error.

HTSeq manual page:
http://www-huber.embl.de/users/anders/HTSeq/doc/count.html
