Title: Past Year as a Junior Bioinformatician in Perspective
Date: 2015-09-19 10:53
Author: monsterbashseq
Category: Bioinformatics, Data Analyses, Personal, Sequencing
Slug: past-year-as-a-junior-bioinformatician-in-perspective
Status: published

Starting a new chapter in life this week attending grad school (again)
towards a PhD. Before moving on, wanted to put into perspective and
share a few notes on what I've learned over the past year working as a
junior bioinformatician for a service core. It's been a long journey,
was a difficult decision to leave, and have moved several different
institutions and cities now. I am focusing and following my own research
interests. In the long run, I'll be of better use to people this way.
Looking forward to the next steps!

-   I got the job by responding to a [post on the Bioconductor
    listserv](https://support.bioconductor.org/p/56545/). I had an
    interdisciplinary-enough background coming with MS research
    degree, 4 years of molecular bench work experience, some makeshift
    computational/programming skills, attended excellent [Workshop on
    Genomics](http://evomics.org/), [Software
    Carpentry](http://wltrimbl.github.io/2013-11-14-woodshole/)
    workshop, and was really interested. They were willing to train me.
-   They had foresight to hire junior-level positions. My colleagues and
    I did the more routine stuff like Illumina demultiplexing and fairly
    well-worked out pipelines, e.g. RNAseq so the higher-level
    bioinformaticians could be free to tackle larger research projects
    and administrative challenges.
-   Worked on a side project benchmarking *de novo* transcriptome
    assemblers, ended up being in the right place at the right time, and
    wrote a review chapter for a
    [book](http://cshlpress.com/default.tpl?cart=144267682315232380&action=full&--eqskudatarq=1041).
-   Learned to reach out to people I admire and ask questions.
-   "[The perfect is the enemy of the
    good](https://en.wikipedia.org/wiki/Perfect_is_the_enemy_of_good)."
    (or as someone I met recently put it, why does it have to be good,
    can't it be excellent???) It's never going to be perfect. Do your
    best to get results out, otherwise they will sit with you on and on
    for continued time.
-   I worked on data analyses for 17 different PI in the past 12 months.
    Some in more depth than others, but ultimately had to learn about 17
    different research goals/questions/designs. That was just me. My
    colleagues did even more than I did.
-   There's no substitute for working in the same office as two nice,
    smart people working on similar things. Or walking across the hall
    and asking my supervisor, having him spend the time to outline an
    answer or discussion so that we both understand more about what's
    going on. I've motivated a lot of self-guided study over the past
    few years, learning to [seek
    help](http://www.catb.org/esr/faqs/smart-questions.html) through the
    online community when I was isolated in a place where no one else
    was doing what I was doing. But, if there's a problem that I've been
    staring at for the past 20 min-to-an-hr, having people physically
    present and just asking "Can you take a look at this?" can save a
    lot of time and has really escalated my understanding and grasp of
    bioinformatics. Plus, it's fun to laugh together when service
    clients send funny emails. **I am eternally grateful for my former
    officemates, supervisors and all of my colleagues.**
-   There was genuine interest and initiative for improving relationship
    between biomedical research community and bioinformatics support.
    High demands for diverse services create challenges and constraints
    on time.
-   There are benefits to being at a large institution (I had come from
    a small academic institution). If someone doesn't know the answer to
    your question, someone else might. Or, different conversations with
    different people will be thought-provoking towards a different way
    of looking at things. Very supportive, diverse and productive
    atmosphere.
-   Co-author on 2 papers,
    [RNAseq](http://onlinelibrary.wiley.com/doi/10.1002/stem.1921/abstract?userIsAuthenticated=false&deniedAccessCustomisedMessage=)
    and [16S community profiling
    analyses](http://www.sciencedirect.com/science/article/pii/S0960894X14013444)
-   That's not to say there aren't politics associated. There will be
    negatives about every place.
-   At the end of the day, you have to be happy with both what you're
    doing and where you live.
-   In a service-oriented facility, some PI just want results. I've
    experienced this stalemate first hand:

https://www.flickr.com/photos/lpcohen/21355199539/in/dateposted-public/

-   Bioinformatics is a process, [way of
    life](https://biomickwatson.wordpress.com/2013/08/06/bioinformatics-is-not-something-you-are-taught-its-a-way-of-life/),
    a field of study in itself, and not just a set of software tools
    that will output results. Requires hypotheses and protocols and
    being [skeptical of your
    results](http://www.molecularecologist.com/2015/04/dont-trust-your-data-reviewing-bioinformatics-data-skills/)
    then going back and doing it again to make sure you can reproduce.
    What I love about bioinformatics is that you can go back and do it
    again in a slightly different way to see what happens, testing what
    parameter caused the interesting results. Not like in the lab where
    reagents are SO expensive, samples hard to come by, that if you were
    to go back and do it again, it would double your costs. With
    bioinformatics, there are costs but in time and computing resources.
-   I'm convinced that results and interpretations of bioinformatics
    tools are better if the PI or grad student or someone deeply
    involved in the research question learns about the tools, can
    program in R or use the commandline tools themselves. I really think
    that [every biologist should learn how to
    program](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1000589),
    but I realize that's not realistic expectation. So, at the very
    least, if bioinformatics results are a necessary part of the study,
    students, post-docs, someone in the team, should be active in
    learning about how to use the bioinformatics tools.
-   I knew stuff about microarrays and was willing to analyze microarray
    data. People still have microarray data and seemed to be grateful.
    The field seems to be moving away from microarrays. Yes, data are
    messy, noisy. Don't give you unsolicited sequence information like
    RNAseq. Cost of microarray vs. RNAseq might even be higher at this
    point because of analyses compared to RNAseq. But, I'm not convinced
    that this is a completely dead technology that should be discarded.
    Might be more appropriate data type for some questions. Such as...?
-   I'm glad that I postponed grad school over this year rather than
    accepting admissions offers last year. It gave me a chance to
    explore more in depth, and meet new people. Question the grad school
    process and even apply to a few more schools. (Another post
    forthcoming about grad school application process.) Now is
    definitely the time. Answer to '[Do you really want a
    PhD?](http://shouldigetaphd.com/)' is now a resounding "Yes!" Can't
    imagine doing anything else. Excited about future.
-   I still feel like a gigantic moron most of the time, write code
    badly in a way that is not [collaborative or
    sustainable](https://mozillascience.github.io/codeReview/intro.html).
    Sometimes I find that I can't reproduce my own analyses several
    months later and then end up having to re-do past work. (Everything
    is documented, just in a haphazard way that makes sense for me, but
    not necessarily someone else - or my future self in 3 months.) Need
    to work on all of that that. [Session on reproducibility at
    NGS2015](https://monsterbashseq.wordpress.com/2015/08/25/reproducible-research-using-rmarkdown-ngs2015-week-3/)
    helped tremendously as does github.
-   I still remember the first time I got a big HPC job running, and saw
    my name on the board. It was exciting! Don't forget that:

https://www.flickr.com/photos/lpcohen/14243536613/in/dateposted-public/
