Title: De novo transcriptome assembly, annotation and evaluation of the branching tube Caribbean sponge, Aiolochroia crassa
Date: 2018-10-20
Category: transcriptomes
Tags: sequencing, sponge, RNAseq, transcriptomes
Author: Lisa K. Johnson, Sara Edge
Summary: Sharing and seeking advice on a *de novo* transcriptome assembly and raw RNAseq data 

Authors: Lisa K. Johnson and Sara Edge

This blog post was inspired by attending the [Global Invertebrate Genomics Alliance (GIGA) conference](https://gigaiii.weebly.com/) in Curaçao (Oct. 19-21, 2018). There is a [paucity of invertebrate sequencing information available](https://academic.oup.com/jhered/article/105/1/1/858593), especially for Poriferans. Invertebrate genomes and transcriptomes allow us to better understand global biodiversity towards conservation efforts.

This is raw RNA-seq data and an assembled transcriptome that has been sitting on servers since 2013. We want to release it so that perhaps this can help anyone who is interested.

Writing manuscripts for publication is time consuming. Submitting raw reads and transcriptomes to ENA or NCBI, SRA and TSA is also time consuming and confusing. Since this is an old project that is not related to my dissertation research, instead of spending valuable research time on these efforts, we make these data and information available.

Link to the data: [https://doi.org/10.17605/OSF.IO/B972E](https://doi.org/10.17605/OSF.IO/B972E)

Citation:
```
Johnson, L., & Edge, S. (2018, August 13). De novo transcriptome of the branching tube Caribbean sponge, Aiolochroia crassa. https://doi.org/10.17605/OSF.IO/B972E
```

Eventually, we would like to write and submit. GIGA provides a [central resource](http://giga-cos.org/) for submitting these data as well. I should probably release these data there as well. In the meantime, here you go! If you are interested, please get in touch. Advice and comments welcome below for the time when we do work towards a publishable manuscript. In particular, what additional information, analysis and discussion would be needed with these data that would be required for an original and useful publication in this field? 

## Special thanks 

This work was done at [FAU-HBOI](http://www.fau.edu/hboi/) in 2012 when I worked for the [Robertson Coral Reef Research Program](http://www.fau.edu/hboi/meh/crrobertson.php). [Ameer Tohamy](https://www.ratemyteachers.com/ameer-ahmed-tohamy/7315033-t) and Thomas Camacho provided assistance while conducting the exposure experiment in 2012. [Dr. Joe Lopez](https://cnso.nova.edu/lopez/microlab/About%20Dr.Lopez.html) provided logistics and assistance in arranging for the NSEU diving boat for sample collections. [Dr. Shirley Pomponi](http://www.fau.edu/hboi/mbbr/spongebio.php) for sample identification and consultation with the experiment. [Dr. Joshua Voss](http://vosslab.weebly.com/dr-joshua-voss.html) provided support while finishing out the project in 2013. Funding for this project and sequencing was provided to [Dr. Sara Edge](https://scholar.google.com/citations?user=E-Ktuu8AAAAJ&hl=en) through the FAU-HBOI Foundation ['Save Our Seas' specialty license plate](https://www.flhsmv.gov/dmv/specialtytags/environmental/Save_Our_Seas.html) funds awarded in June 2011. 

Thank you to the Moore Foundation Data-Driven-Discovery award to my advisor, [Dr. C. Titus Brown](http://ivory.idyll.org/lab/) at UC Davis, for funding my time while I am currently working on my PhD, towards projects related to open science, reproducibility and data sharing. Thank you to Titus and the DIB lab for advice, support and development of the [Eel Pond Protocol](https://khmer-protocols.readthedocs.io/en/latest/mrnaseq/index.html), [old version](https://eel-pond.readthedocs.io/en/latest/) and [new (with snakemake, by Dr. Tessa Pierce](https://github.com/dib-lab/eelpond).

# Summary
* [*Aiolochroia crassa*](https://doi.org/10.17605/OSF.IO/B972E) is a common sponge species found on FL reefs
* Samples from six colonies of *Aiolochroia crassa* collected off Fort Lauderdale, FL in 2012
* Samples were fragmented and exposed (in a balanced experimental design) to an acute 40 hrs challenge in 100 ppm crude WAF (Water accomodated fraction) oil and 10 ppm dispersant (Corexit 9500A, Nalco)
* Sponge cDNA samples were hybridized to a cDNA coral microarray (n=72)
* RNAseq data from pooled samples (n=18) in each treatment group (4 groups)
* *De novo* transcriptome assembly and annotation was performed using the [eel pond protocol](https://eel-pond.readthedocs.io/en/latest/), [files available](https://osf.io/b972e/)

Download all files via commandline:
```
pip install osfclient
osf -p b972e clone A_crassa
```

* This sequencing effort adds to the [sparse sequencing information available from this species](https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=261677)
 

![](https://i.imgur.com/rOZjgng.png)

# Introduction

Porifera are among the major phyla inhabiting marine hard substrate benthos, in terms of number of species and biomass (Carballo et al. 1996). As sessile filter-feeders, sponges are sensitive to stressors and thus are important indicator species of environmental perturbations (Carballo et al., 1996; Bachinski et al., 1997).

Gene expression profiling has been used as a method for identifying suites of genes in functional pathways expressed under certain controlled conditions in reef-building corals (Edge et al. 2005) and verified as a tool to detect similar sub-lethal responses from ex situ environmental coral samples (Edge et al. 2012, Edge et al. 2008). Applying similar gene expression profiling methods to sponges under controlled conditions, such as exposure to oil and dispersant, could assist resource managers in monitoring the effects of these stressors on reef sponges. 

The experiment aimed to assess the responses of a single sponge species following exposure to oil with a similar composition to the MC252 Macondo wellhead after the Deepwater Horizon event and to the dispersant used during clean-up efforts.

A reference *de novo* transcriptome was assembled and annotated for *Aiolochroia crassa*. This sequence information is the first from this species (aside from [25 NCBI nucleotide records](https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=261677)). 

**TO DO: check to see if these genes are repsented in the transcriptome**.


# Methods

Fragments from six *Aiolocroia crassa* sponge colonies were collected from a Ft. Lauderdale reef and transported to the aquaculture facility at Harbor Branch Oceanographic Institute at Florida Atlantic University within two hours of collection (29 May 2012). Sponges were maintained in aerated raceways with filtered and UV-sterilized seawater (32 ppt, 28◦C) pumped from the Indian River Lagoon. Each sponge colony was cut into twelve pieces (4-6 cm2) resulting in 72 fragments. Colored plastic zip ties were used to identify sponge colonies and secure fragments to twelve 20 cm by 45 cm PVC grids (Figure 1). 


| Treatment | Description | N |
| --------- | ----------- | -- |
| OD |  oil, dispersant | 18 |
| OC |  oil, no dispersant | 18 |
| UD| no oil, dispersant | 18 |
| UC| no oil, no dispersant | 18 |


![](https://i.imgur.com/uU2HFFY.png)
Figure 1. Sponge fragments from each parent colony attached to PVC grid and assigned a unique colored cable tie for identification. One grid per tank x 12 tanks, four treatment groups x 3 replicates per treatment.

### Experimental Exposures
After 24 hours of acclimation, fragmented sponges were placed into twelve 5 gallon aquaria filled with 15 L of UV-treated seawater. Aquaria were randomly distributed between two aquaculture raceways. There were four treatment groups, each replicated three times, in a randomized block experimental design.

Treatments consisted of exposure to 100 ppm oil and 10 ppm dispersant (Corexit 9500A, Nalco) (OD), oil and no dispersant (OC), dispersant and no oil (UD) and a control (UC). Oil used for the experiment was a fresh oil/source sample collected by Cardno ENTRIX for British Petroleum from near the MS252 Macondo wellhead (Coordinates: 28.866014, -88.056264, Sample ID: SO-20110212-HMPA4-003) on 12 February 2011. A 3,000 ppm stock solution of oil was made by mixing with sea water and stirring continuously on low heat for 18 hrs. Oil concentration was based on previously-published experiments (Epstein et al. 2000, Dodge et al 1984, Cook and Knap 1983) and dispersant concentration was based on the Environmental Protection Agency’s recommended application range (dispersant to oil ratio of 1:50 to 1:10, EPA).

Environmental conditions were consistent throughout the exposures. Salinity was adjusted to 32 ppt in each tank using Instant Ocean® while temperature was maintained at approximately 28◦C. During the exposure period, tank chemistry, including dissolved oxygen, pH, salinity, and temperature, were recorded three times daily. Additionally, 80% of the water in all aquaria was replaced and the same concentrations of oil and dispersant, or seawater only, were added 15 hours after the initial exposure. Sponges in each tank were photographed twice daily and monitored for visible signs of stress. After 40 hours, all samples were collected, preserved in TRIzol nucleic acid preservative (Invitrogen, Inc.) and frozen at -80◦C for molecular analyses.

### Microarray

A custom stress-focused microarray with 1,695 scleractinian coral genes divided into 51 functional groups was used to assess gene expression profiles (Agilent Technologies). Gene functions range from normal cellular responses, such as metabolism, signaling and protein binding, to a suite of responsive genes related to biotransformation, apoptosis, DNA damage and repair, xenobiotics, drug metabolism, oxidative stress and antioxidant defense. Probes on the microarray consist of 60 base pair oligonucleotides that are representative of coral gene exon open reading frames (ORF) coding for a specific protein. The coral microarray can be used with sponge samples to test the similarity between these metazoan species.

### Sample processing for molecular analyses

Total RNA was extracted from each sponge sample, reverse transcribed then copy RNA was synthesized and labeled with Cy3 dye following the manufacturer’s protocol (Agilent Technologies, Catalogue# 5190-2305). All Cy3-labeled cRNA samples were hybridized to individual microarrays using methods described in the manufacturer’s protocol (Agilent Technologies). Microarray chips were hybridized at 65ºC for 17 hrs then washed using solutions from Agilent. Chips were scanned and imaged at 535 nm using a high-resolution fluorescent GenePix® 4200A microarray scanner (Axon, Molecular Devices, California, USA).  

### RNA-Seq

Total RNA samples from each colony were combined by treatment group to yield 10 µg at a concentration of >200 ng/µL. The quality and quantity of the pooled samples were verified on a Nanodrop 1000 and Agilent Bioanalyzer 2100. RNA samples were shipped on dry ice to BaseClear (Netherlands) in November 2012 for library preparation and sequencing on an Illumina HiSeq 2000.

### Microarray Data analyses

Raw intensity data were extracted from scanned images using GenePixPro 6.0 software (Molecular Devices) and imported into the R statistical computing environment. LIMMA (Linear Models for Microarray Data) was used for microarray analysis and data  quality was assessed using arrayQualityMetrics. 



# Computational resources

* NSF-XSEDE Jetstream cloud computing resources were used (allocation TG-BIO160028 to LKJ)
* s1.xxlarge (CPU: 44, Mem: 120 GB, Disk: 480 GB, Disk: 480 GB root)
* Ubuntu 16.04 Devel and Docker v1.13
* Trinity v2.6.6 Patch Release
* [Eel Pond protocol](http://eel-pond.readthedocs.io/en/latest/)

# Results

### Data files storage:

* OSF respository: https://osf.io/b972e/

```
DOI 10.17605/OSF.IO/B972E
```
### Raw reads, trimming

* TruSeq Poly-A library
* PEx50

![](https://i.imgur.com/AFq5yvv.png)

### Diginorm

```
read 231786047 reads, 11815700653 bp
wrote 67649858 reads, 3433851077 bp
looked at 63961781 reads twice (1.28 passes)
removed 164136189 reads and trimmed 3055905 reads (72.13%)
trimmed or removed 70.94%% of bases (8381849576 total)
231786047 reads were high coverage (100.00%);
skipped 0 reads/0 bases because of low coverage
fp rate estimated to be 0.019
output streamed to stdout
DONE; read 67649858 sequences, 32964374 pairs and 1721110 singletons
wrote to: paired.fq.gz and single.fq.gz
```

```
DONE; split 65928748 sequences (32964374 left, 32964374 right, 0 orphans)
/1 reads in paired.fq.gz.1
/2 reads in paired.fq.gz.2
```

### Assembly

Total reads for assembly:
```
58,354,750
```
Trinity version: Trinity-v2.6.6
```
Statistics:
===========
Trinity Version:      Trinity-v2.6.6
Compiler:             GCC
Trinity Parameters:   --left left.fq --right right.fq --seqType fq --max_memory 14G --CPU 16
Paired mode
 Input data
  Left.fasta    2903 MByte
  Right.fasta   2902 MByte
  Number of unique KMERs: 419308215
  Number of reads:        58354750 Output data
  Trinity.fasta 80 MByte

Runtime
=======
Start:       Sat May  5 21:19:46 EDT 2018
End:         Sun May  6 08:06:56 EDT 2018
Trinity   38830 seconds
  Inchworm (phase 1 - read clustering)  2967 seconds
  Chrysalis (phase 1 - read clustering) 33642 seconds
  Rest (phase 2 - parallel assembly)       2221 seconds
```

Trinity new version:
```
[ INFO] 2018-05-06 21:52:20 : Loading assembly: /home/ljcohen/A_crassa/assembly/trinity_out_dir/Trinity.fasta
[ INFO] 2018-05-06 21:52:36 : Analysing assembly: /home/ljcohen/A_crassa/assembly/trinity_out_dir/Trinity.fasta
[ INFO] 2018-05-06 21:52:36 : Results will be saved in /home/ljcohen/A_crassa/assembly/A_crassa_transrate/Trinity
[ INFO] 2018-05-06 21:52:36 : Calculating contig metrics...
[ INFO] 2018-05-06 21:52:51 : Contig metrics:
[ INFO] 2018-05-06 21:52:51 : -----------------------------------
[ INFO] 2018-05-06 21:52:51 : n seqs                       119109
[ INFO] 2018-05-06 21:52:51 : smallest                        201
[ INFO] 2018-05-06 21:52:51 : largest                        9126
[ INFO] 2018-05-06 21:52:51 : n bases                    70503595
[ INFO] 2018-05-06 21:52:51 : mean len                     591.93
[ INFO] 2018-05-06 21:52:51 : n under 200                       0
[ INFO] 2018-05-06 21:52:51 : n over 1k                     16468
[ INFO] 2018-05-06 21:52:51 : n over 10k                        0
[ INFO] 2018-05-06 21:52:51 : n with orf                    37747
[ INFO] 2018-05-06 21:52:51 : mean orf percent              82.28
[ INFO] 2018-05-06 21:52:51 : n90                             262
[ INFO] 2018-05-06 21:52:51 : n70                             458
[ INFO] 2018-05-06 21:52:51 : n50                             787
[ INFO] 2018-05-06 21:52:51 : n30                            1373
[ INFO] 2018-05-06 21:52:51 : n10                            2748
[ INFO] 2018-05-06 21:52:51 : gc                             0.52
[ INFO] 2018-05-06 21:52:51 : bases n                           0
[ INFO] 2018-05-06 21:52:51 : proportion n                    0.0

```
Trinity 2014 version

```
[ INFO] 2018-05-06 21:55:00 : Loading assembly: /home/ljcohen/baseclear/sponge_Trinity_old.fasta
[ INFO] 2018-05-06 21:55:13 : Analysing assembly: /home/ljcohen/baseclear/sponge_Trinity_old.fasta
[ INFO] 2018-05-06 21:55:13 : Results will be saved in /home/ljcohen/baseclear/transrate_results/sponge_Trinity_old
[ INFO] 2018-05-06 21:55:13 : Calculating contig metrics...
[ INFO] 2018-05-06 21:55:24 : Contig metrics:
[ INFO] 2018-05-06 21:55:24 : -----------------------------------
[ INFO] 2018-05-06 21:55:24 : n seqs                        95532
[ INFO] 2018-05-06 21:55:24 : smallest                        201
[ INFO] 2018-05-06 21:55:24 : largest                       14515
[ INFO] 2018-05-06 21:55:24 : n bases                    57208347
[ INFO] 2018-05-06 21:55:24 : mean len                     598.84
[ INFO] 2018-05-06 21:55:24 : n under 200                       0
[ INFO] 2018-05-06 21:55:24 : n over 1k                     13778
[ INFO] 2018-05-06 21:55:24 : n over 10k                        3
[ INFO] 2018-05-06 21:55:24 : n with orf                    30120
[ INFO] 2018-05-06 21:55:24 : mean orf percent              84.06
[ INFO] 2018-05-06 21:55:24 : n90                             252
[ INFO] 2018-05-06 21:55:24 : n70                             456
[ INFO] 2018-05-06 21:55:24 : n50                             860
[ INFO] 2018-05-06 21:55:24 : n30                            1558
[ INFO] 2018-05-06 21:55:24 : n10                            3151
[ INFO] 2018-05-06 21:55:24 : gc                             0.53
[ INFO] 2018-05-06 21:55:24 : bases n                           0
[ INFO] 2018-05-06 21:55:24 : proportion n                    0.0
[ INFO] 2018-05-06 21:55:24 : Contig metrics done in 12 seconds
[ INFO] 2018-05-06 21:55:24 : No reads provided, skipping read diagnostics
[ INFO] 2018-05-06 21:55:24 : No reference provided, skipping comparative diagnostics
[ INFO] 2018-05-06 21:55:24 : Writing contig metrics for each contig to /home/ljcohen/baseclear/transrate_results/sponge_Trinity_old/contigs.csv
[ INFO] 2018-05-06 21:55:28 : Writing analysis results to assemblies.csv
```
#### BUSCO

Benchmarking universal single-copy orthologs (BUSCO) is a metric for assessing the completeness of a transcriptome against databases of genes expected to be found in a group of species [Simão et al. 2015](https://academic.oup.com/bioinformatics/article/31/19/3210/211866).

This *A. crassa* transcriptome had an approximately 75% complete BUSCO scores compared to both the metazoan and eukaryota databases.

Metazoa
```
# BUSCO version is: 3.0.2 
# The lineage dataset is: metazoa_odb9 (Creation date: 2016-02-13, number of species: 65, number of BUSCOs: 978)
# To reproduce this run: python /home/ljcohen/busco/scripts/run_BUSCO.py -i /home/ljcohen/A_crassa/assembly/A_crassa.Trinity.fasta -o A_crassa.busco.metazoa -l /home/ljcohen/reference/metazoa_odb9/ -m transcriptome -c 4
#
# Summarized benchmarking in BUSCO notation for file /home/ljcohen/A_crassa/assembly/A_crassa.Trinity.fasta
# BUSCO was run in mode: transcriptome

	C:75.0%[S:47.3%,D:27.7%],F:12.4%,M:12.6%,n:978

	734	Complete BUSCOs (C)
	463	Complete and single-copy BUSCOs (S)
	271	Complete and duplicated BUSCOs (D)
	121	Fragmented BUSCOs (F)
	123	Missing BUSCOs (M)
	978	Total BUSCO groups searched
```
Eukaryota
```
# BUSCO version is: 3.0.2 
# The lineage dataset is: eukaryota_odb9 (Creation date: 2016-11-02, number of species: 100, number of BUSCOs: 303)
# To reproduce this run: python /home/ljcohen/busco/scripts/run_BUSCO.py -i /home/ljcohen/A_crassa/assembly/A_crassa.Trinity.fasta -o A_crassa.euk -l /home/ljcohen/reference/eukaryota_odb9/ -m transcriptome -c 4
#
# Summarized benchmarking in BUSCO notation for file /home/ljcohen/A_crassa/assembly/A_crassa.Trinity.fasta
# BUSCO was run in mode: transcriptome

	C:75.2%[S:42.2%,D:33.0%],F:16.2%,M:8.6%,n:303

	228	Complete BUSCOs (C)
	128	Complete and single-copy BUSCOs (S)
	100	Complete and duplicated BUSCOs (D)
	49	Fragmented BUSCOs (F)
	26	Missing BUSCOs (M)
	303	Total BUSCO groups searched
```

### Annotation

* [Used protein .fa from *Amphimedon queenslandica* on NCBI, RefSeq](ftp://ftp.ncbi.nlm.nih.gov/genomes/Amphimedon_queenslandica/protein/)
* 75.6% of contigs (90,047 out of 119,109) had annotations
* 59,136 annotations from *Amphimedon queenslandica* including isoforms.
* Filtering those with E-value < 1e-05, dropping those with "NA" in gene name and choosing only one gene name per contig with top E-value score = 63,084 annotations with 1,910 (3%) from *Amphimedon queenslandica*.


### Quantification

With salmon v0.9.1. Files are here:
https://osf.io/b972e/

```
./UC_CTTGTA.quant/aux_info/meta_info.json:    "percent_mapped": 77.32713084784302,
./OC_GCCAAT.quant/aux_info/meta_info.json:    "percent_mapped": 79.4706051208496,
./OD_CGATGT.quant/aux_info/meta_info.json:    "percent_mapped": 79.15476656959088,
./UD_TGACCA.quant/aux_info/meta_info.json:    "percent_mapped": 79.85752052382412,
```


### Comparison with the coral microarray

Assembly of these data with the Trinity software package, 2014 version had fewer hits to the microarray than the latest version in 2018.
```
makeblastdb -in sponge_Trinity_old.fasta -dbtype nucl
blastn -query Agilent_microarray.fasta -db sponge_Trinity_old.fasta -out microarray_v_Trinity -evalue 1e-5 -outfmt 6 -max_target_seqs 1
blastn -query ../microarray/032951_1467726033734/FASTA\032951_D_Fasta_20110309.txt -db sponge_Trinity_old.fasta -out microarray_v_Trinity -evalue 1e-5 -outfmt 6 -max_target_seqs 1
```

```
ljcohen@js-156-111:~$ cat  Trinity_v_microarray
comp789632_c0_seq1	CUST_3_PI426266271	100.00	29	0	0	641	669	17	45	8e-09	54.7
comp936137_c4_seq3	CUST_33_PI426226915	94.83	58	3	0	714	771	58	1	6e-20	91.6
comp936137_c4_seq4	CUST_33_PI426226915	94.83	58	3	0	719	776	58	1	6e-20	91.6
comp936137_c4_seq5	CUST_33_PI426226915	94.83	58	3	0	714	771	58	1	6e-20	91.6
comp937294_c5_seq3	CUST_3_PI426226915	100.00	35	0	0	92	126	10	44	1e-12	65.8
comp937846_c1_seq1	CUST_372_PI426266615	93.33	60	4	0	2516	2575	60	1	6e-19	89.8
ljcohen@js-156-111:~$ cat microarray_v_Trinity 
CUST_371_PI426266615	comp937846_c1_seq1	93.33	60	4	0	1	60	2574	2515	2e-18	89.8
CUST_3_PI426226915	comp937294_c5_seq3	100.00	35	0	0	10	44	92	126	3e-11	65.8
CUST_3_PI426266271	comp789632_c0_seq1	100.00	29	0	0	17	45	641	669	7e-08	54.7
CUST_370_PI426266615	comp937846_c1_seq1	93.33	60	4	0	1	60	2571	2512	2e-18	89.8
CUST_33_PI426226915	comp936137_c4_seq5	94.83	58	3	0	1	58	771	714	6e-19	91.6
CUST_372_PI426266615	comp937846_c1_seq1	93.33	60	4	0	1	60	2575	2516	2e-18	89.8
```

Trinity 2.6.6
```
makeblastdb -in /home/ljcohen/A_crassa/assembly/A_crassa.Trinity.fasta -dbtype nucl
blastn -query /home/ljcohen/A_crassa/assembly/A_crassa.Trinity.fasta -db /home/ljcohen/microarray/032951_1467726033734/Agilent_microarray.fasta -out microarray_v_Trinity.2.6.6 -evalue 1e-5 -outfmt 6 -max_target_seqs 1
blastn -query /home/ljcohen/microarray/032951_1467726033734/Agilent_microarray.fasta -db /home/ljcohen/A_crassa/assembly/A_crassa.Trinity.fasta -out Trinity.2.6.6_v_microarray -evalue 1e-5 -outfmt 6 -max_target_seqs 1
```
```
ljcohen@js-156-111:~/A_crassa/assembly$ cat Trinity.2.6.6_v_microarray
CUST_395_PI426266615	TRINITY_DN48401_c7_g1_i10	93.18	44	2	1	12	55	5	47	1e-10	63.9
CUST_301_PI426266615	TRINITY_DN49097_c3_g1_i20	89.47	57	5	1	5	60	184	240	9e-13	71.3
CUST_396_PI426266615	TRINITY_DN48401_c7_g1_i10	93.18	44	2	1	13	56	5	47	1e-10	63.9
CUST_371_PI426266615	TRINITY_DN48711_c1_g1_i3	96.67	60	2	0	1	60	300	359	1e-21	100
CUST_3_PI426226915	TRINITY_DN49097_c3_g1_i5	98.11	53	1	0	1	53	224	276	2e-19	93.5
CUST_12_PI426227955	TRINITY_DN47826_c2_g1_i1	94.74	38	2	0	12	49	74	111	2e-09	60.2
CUST_3_PI426266271	TRINITY_DN36002_c0_g1_i1	100.00	29	0	0	17	45	641	669	9e-08	54.7
CUST_394_PI426266615	TRINITY_DN48401_c7_g1_i10	93.18	44	2	1	11	54	5	47	1e-10	63.9
CUST_370_PI426266615	TRINITY_DN48711_c1_g1_i3	96.67	60	2	0	1	60	303	362	1e-21	100
CUST_33_PI426226915	TRINITY_DN50786_c0_g2_i3	94.83	58	3	0	1	58	120	63	7e-19	91.6
CUST_372_PI426266615	TRINITY_DN48711_c1_g1_i3	96.67	60	2	0	1	60	299	358	1e-21	100
CUST_106_PI426266615	TRINITY_DN45854_c4_g1_i1	90.00	60	6	0	1	60	130	189	5e-15	78.7
CUST_302_PI426266615	TRINITY_DN49097_c3_g1_i20	89.29	56	5	1	6	60	184	239	3e-12	69.4
ljcohen@js-156-111:~/A_crassa/assembly$ cat microarray_v_Trinity.2.6.6
TRINITY_DN49097_c3_g1_i18	CUST_3_PI426226915	98.11	53	1	0	144	196	1	53	3e-21	93.5
TRINITY_DN49097_c3_g1_i5	CUST_3_PI426226915	98.11	53	1	0	224	276	1	53	4e-21	93.5
TRINITY_DN49097_c3_g1_i15	CUST_301_PI426266615	89.47	57	5	1	184	240	5	60	3e-14	71.3
TRINITY_DN49097_c3_g1_i20	CUST_301_PI426266615	89.47	57	5	1	184	240	5	60	3e-14	71.3
TRINITY_DN49097_c3_g2_i1	CUST_3_PI426226915	95.45	44	2	0	96	139	10	53	2e-14	71.3
TRINITY_DN49097_c3_g2_i3	CUST_3_PI426226915	95.45	44	2	0	96	139	10	53	2e-14	71.3
TRINITY_DN45854_c4_g1_i1	CUST_106_PI426266615	90.00	60	6	0	130	189	1	60	7e-17	78.7
TRINITY_DN47826_c2_g1_i1	CUST_12_PI426227955	94.74	38	2	0	74	111	12	49	3e-11	60.2
TRINITY_DN47776_c2_g1_i2	CUST_394_PI426266615	97.14	35	1	0	209	243	23	57	4e-11	60.2
TRINITY_DN47776_c2_g1_i4	CUST_394_PI426266615	97.14	35	1	0	209	243	23	57	4e-11	60.2
TRINITY_DN47776_c2_g1_i3	CUST_394_PI426266615	97.14	35	1	0	245	279	23	57	5e-11	60.2
TRINITY_DN48711_c1_g1_i2	CUST_372_PI426266615	93.33	60	4	0	299	358	1	60	3e-19	89.8
TRINITY_DN48711_c1_g1_i3	CUST_372_PI426266615	96.67	60	2	0	299	358	1	60	5e-23	100
TRINITY_DN48711_c1_g1_i6	CUST_372_PI426266615	93.33	60	4	0	299	358	1	60	7e-20	89.8
TRINITY_DN48711_c1_g1_i7	CUST_372_PI426266615	93.33	60	4	0	299	358	1	60	1e-19	89.8
TRINITY_DN48711_c1_g1_i8	CUST_372_PI426266615	93.33	60	4	0	299	358	1	60	1e-19	89.8
TRINITY_DN48401_c7_g1_i10	CUST_394_PI426266615	93.18	44	2	1	5	47	11	54	3e-12	63.9
TRINITY_DN51186_c1_g1_i1	CUST_33_PI426226915	94.83	58	3	0	600	657	58	1	5e-20	91.6
TRINITY_DN51186_c1_g1_i4	CUST_33_PI426226915	94.83	58	3	0	605	662	58	1	5e-20	91.6
TRINITY_DN51186_c1_g1_i2	CUST_33_PI426226915	94.83	58	3	0	600	657	58	1	4e-20	91.6
TRINITY_DN51186_c1_g1_i11	CUST_33_PI426226915	94.83	58	3	0	605	662	58	1	4e-20	91.6
TRINITY_DN51186_c1_g1_i15	CUST_33_PI426226915	94.83	58	3	0	616	673	58	1	6e-20	91.6
TRINITY_DN51186_c1_g1_i14	CUST_33_PI426226915	94.83	58	3	0	600	657	58	1	6e-20	91.6
TRINITY_DN51186_c1_g1_i13	CUST_33_PI426226915	94.83	58	3	0	616	673	58	1	6e-20	91.6
TRINITY_DN51186_c1_g1_i18	CUST_33_PI426226915	94.83	58	3	0	616	673	58	1	6e-20	91.6
TRINITY_DN51186_c1_g1_i9	CUST_33_PI426226915	94.83	58	3	0	605	662	58	1	7e-20	91.6
TRINITY_DN51186_c1_g1_i19	CUST_33_PI426226915	94.83	58	3	0	605	662	58	1	7e-20	91.6
TRINITY_DN51186_c1_g1_i20	CUST_33_PI426226915	94.83	58	3	0	370	427	58	1	2e-20	91.6
TRINITY_DN51186_c1_g1_i17	CUST_33_PI426226915	94.83	58	3	0	600	657	58	1	6e-20	91.6
TRINITY_DN51186_c1_g1_i6	CUST_33_PI426226915	94.83	58	3	0	605	662	58	1	7e-20	91.6
TRINITY_DN51186_c1_g1_i12	CUST_33_PI426226915	94.83	58	3	0	605	662	58	1	6e-20	91.6
TRINITY_DN36002_c0_g1_i1	CUST_3_PI426266271	100.00	29	0	0	641	669	17	45	8e-09	54.7
TRINITY_DN50786_c0_g1_i15	CUST_33_PI426226915	97.22	36	1	0	1	36	36	1	3e-11	62.1
TRINITY_DN50786_c0_g1_i9	CUST_33_PI426226915	97.22	36	1	0	1	36	36	1	3e-11	62.1
TRINITY_DN50786_c0_g1_i12	CUST_33_PI426226915	97.22	36	1	0	1	36	36	1	1e-11	62.1
TRINITY_DN50786_c0_g2_i5	CUST_33_PI426226915	94.83	58	3	0	132	189	58	1	1e-20	91.6
TRINITY_DN50786_c0_g2_i3	CUST_33_PI426226915	94.83	58	3	0	63	120	58	1	2e-20	91.6
TRINITY_DN49887_c4_g2_i14	CUST_3_PI426226915	100.00	34	0	0	1	34	5	38	4e-12	63.9
```

# References
[The Analysis of Eight Transcriptomes from All Poriferan Classes Reveals Surprising Genetic Complexity in Sponges](https://academic.oup.com/mbe/article/31/5/1102/993377)

[Deep developmental transcriptome sequencing uncovers numerous new genes and enhances gene annotation in the sponge Amphimedon queenslandica](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-015-1588-z)

[The genome of the contractile demosponge Tethya wilhelma and the evolution of metazoan neural signalling pathways](https://www.biorxiv.org/content/biorxiv/early/2017/03/28/120998.full.pdf)

[Gene Expression Dynamics Accompanying the Sponge Thermal Stress Response](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0165368)

[A Large and Consistent Phylogenomic Dataset Supports Sponges as the Sister Group to All Other Animals](https://www.cell.com/current-biology/fulltext/S0960-9822(17)30199-9)

[Elements of a ‘nervous system’ in sponges](http://jeb.biologists.org/content/218/4/581)

[Comparative transcriptome analysis reveals insights into the streamlined genomes of haplosclerid demosponges](https://www.nature.com/articles/srep18774)

[Transcriptome sequencing and delimitation of sympatric Oscarella species (O. carmela and O. pearsei sp. nov) from California, USA](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0183002)

Carballo, JL et al. 1996. Use of marine sponges as stress indicators in marine ecosystems at Algeciras Bay (southern Iberian Peninsula). [Marine Ecology Progress Series. 135:109-122](http://www.int-res.com/abstracts/meps/v135/p109-122/).

[A genomic view of 500 million years of cnidarian evolution](https://www.sciencedirect.com/science/article/pii/S0168952510002088)

## Existing Genome/Transcriptome resources for other sponge species
* [*Amphimedon queenslandica* genome (model sponge species)](https://www.ncbi.nlm.nih.gov/genome/?term=txid400682[Organism:noexp])
* [A. queenslandica transcriptome and annotation resources](http://amphimedon.qcloud.qcif.edu.au/index.html)
* [*Aplysina aerophoba*](https://www.ncbi.nlm.nih.gov/genome/67299)
* [1,360 Porifera SRA records, 475 RNAseq](https://www.ncbi.nlm.nih.gov/sra/?term=txid6040[Organism:exp])
* [Haliclona tubifera](https://www.ncbi.nlm.nih.gov/nuccore/GFAV00000000)
* [transcriptome shotgun sequencing (TSA) porifera](https://www.ncbi.nlm.nih.gov/nuccore/?term=txid6040%5BOrganism%3Aexp%5D+TSA)
* [	Oscarella carmela](https://www.ncbi.nlm.nih.gov/bioproject/230415)
* [only 25 NCBI Nucleotide records for *Aiolochroia crassa*](https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=261677)
* [Reconstruction of Family-Level Phylogenetic Relationships within Demospongiae (Porifera) Using Nuclear Encoded Housekeeping Genes](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3553142/)
    * [aldolase (ALD)](https://www.ncbi.nlm.nih.gov/nuccore/JQ606737)
    * [catalase (CAT)](https://www.ncbi.nlm.nih.gov/nuccore/JQ606713)
    * [EF1alpha](https://www.ncbi.nlm.nih.gov/nuccore/JQ606687)
* [metagenomes collected from sponge](https://trace.ddbj.nig.ac.jp/DRASearch/study?acc=SRP045752)
