Title: Killifish PromethION data sharing
Date: 2018-10-08
Category: data
Tags: sequencing, killifish, Oxford Nanopore Technologies
Author: Lisa K. Johnson
Summary: Sharing PromethION sequencing data from 4 species of *Fundulus* killifish

We have new long-read [ONT PromethION](https://nanoporetech.com/products/promethion) genomic DNA sequencing data from 4 species of killifish! While we're working with it and generating new Illumina PE150 data for each species, we wanted to share this information and data. Since there are not many PromethION instruments out there, to my knowledge, these data are probably some of the first of its kind to be shared, besides [human data](https://www.ebi.ac.uk/ena/data/view/PRJEB26791).

# Questions for the community

The purpose of this blog post is mainly to share our plan, seek advice, and see if anyone is interested in these data. 

**Would anyone be interested in collaborating with us on an [assemblathon](https://academic.oup.com/gigascience/article/2/1/2047-217X-2-10/2656129)-like project with these ONT data to come up with some best-practices? If so, please contact me! ([twitter](https://twitter.com/monsterbashseq) or [email](mailto:ljcohen@ucdavis.edu))**

We're generating these ONT data for the purpose of assembling reference genomes for organisms where we already have a closely-related (5-20 million years diverged) [genome from a sister species, (*Fundulus heteroclitus*)](https://www.ncbi.nlm.nih.gov/genome/?term=txid8078%5Borgn%5D). We're in the process of generating Illumina data to improve the consensus. By [today's standards of genome assembly](https://twitter.com/olgabot/status/1017127050014949381), this project is not the most sophisticated. We're not planning to do anything fancy to improve the scaffolding with optical and chromatin interaction mapping. Just quick and dirty genomes for around $3500 each. This is also by no means a perfect data set by ONT standards. Our DNA samples - from all species - have consistently not been utilizing the pores as efficiently as possible on either MinION or PromethION flowcells, despite using fresh tissue and troubleshooting efforts with different DNA extraction and cleanup methods. Therefore, our yields from ONT sequencing have not been as high as other groups have gotten. But, from our experience, this is what can happen when new species are sequenced with a new technology. 

Despite these shortcomings, 20-30Gb/species ONT data are great to have! Better than Illumina data alone. Since this is a new type of ONT data from the PromethION, we're in unchartered territory here about some things.

* What does one do with the millions of raw fast5 signal files from the PromethION? Our basecalled fastq are 70-100 GB, but the directory of raw fast5 signal files is >600 GB for each run! Wow. To our knowledge, the only software that uses these files to polish the reads is [Nanopolish](https://github.com/jts/nanopolish) by [Jared Simpson's group](https://twitter.com/jaredtsimpson?lang=en). Since the files are so large, I haven't tried using Nanopolish yet. It it a good idea to try? What is the best way to share >600 GB of millions of fast5 files with people who might be interested in using it?
* Data are unfiltered. Should we remove DNA CS lambda phage spike-in before sharing in a public repository? 
* What is the best way to look for reads suspected as errors (>900 Mb)? Should these be removed before sharing in a public repository?
* What is the best workflow and tools to use for assembly with ONT and Illumina data?
* Where is the best place to share these sequencing data? Even though [zenodo has recently upgraded their file size limits to 50 GB](http://help.zenodo.org/), it is not a good place to store these data based on their response to my request:

{% flickr 4303408795 small "zenodo email" %}

**Any other advice, questions, or comments on things that perhaps we are not considering?**

# Our plan from here:

#### 1. Post data to a public nucleic acid repository

[European Nucleotide Archive (ENA)-EBI](https://www.ebi.ac.uk/ena/submit/sequence-submission) or SRA-NCBI?
(I will post links to the records here once they are available.)

What to do with the raw fast5 signal files?

#### 2. Write a data note

Either [F1000 research](https://f1000research.com/for-authors/article-guidelines/data-notes) or [GigaScience](https://academic.oup.com/gigascience/pages/data_note)

#### 3. Use the genome assemblies!
The ultimate goal is to do some comparative genomics analyses between these genomes, use them as references for RNAseq data, and scan for regulatory regions of interest, e.g. [salinity-responsive enhancing elements (OSRE)](http://www.pnas.org/content/114/13/E2729)

# Thank you!
This is a collaborative effort between my two advisors, Dr. Titus Brown and Dr. Andrew Whitehead. (Seriously, I can't believe how fortunate I am to have two such supportive advisors and a willingly collaborative network of lab members and colleagues.) Other contributors to this project have been [Dr. Ruta Madhusudan](http://dnatech.genomecenter.ucdavis.edu/people/) and [Tony Gill](https://twitter.com/tonygill3?lang=en). Special thanks to [instructors at PoreCamp, Texas](http://porecamp.github.io/texas/), [Dr. Charlie Johnson](https://twitter.com/BioMath?lang=en) and [Dr. Richard Metz](https://twitter.com/DNA_hole?lang=en) at [Texas A&M Agrilife Research Sequencing Facility](http://www.txgen.tamu.edu/#aboutus) (where PoreCamp was held) for Illumina NovaSeq data from [*Fundulus olivaceus*](https://twitter.com/nanopore/status/872910510508953605)). THANK YOU to [Dr. David Duvernell at Missouri University of Science & Technology](http://people.mst.edu/faculty/duvernelld/index.html) and [Dr. Jake Schaefer at the University of Southern Mississippi](http://ichthyology.usm.edu/) for sending us all of these fish to us, live! 

All of this is made possible by funding support from the Moore Foundation Data-Driven-Discovery investigator award to Dr. Titus Brown.

Below is a bit of background on the project, methods, preliminary results, references, and a list of some tools I've compiled.

# New PromethION instrument at the UC Davis Genome Center
In [May 2018](https://twitter.com/meganamsu/status/997216050247864322), the [UC Davis Genome Center](http://dnatech.genomecenter.ucdavis.edu/) received its PromethION from Oxford Nanopore! This was part of a collaborative effort to acquire this instrument through the [PromethION early release program (PEAP)](https://www.genomeweb.com/sequencing-technology/early-access-oxford-nanopore-promethion-requires-75k-deposit-24k-access-fee#.W70sSpNKhTY) back in early 2016 by Genome Center faculty: [Dr. Richard Michelmore](https://biology.ucdavis.edu/people/richard-michelmore), [Dr. Megan Dennis](https://biology.ucdavis.edu/people/megan-dennis) (who is on my dissertation committee!), [Dr. Luis Carvajal-Carmona](https://biology.ucdavis.edu/people/luis-g-carvajal-carmona), and [Dr. Titus Brown](http://genomecenter.ucdavis.edu/people/faculty/name/c-titus-brown/).

{% flickr 42629095711 small "PromethION at UC Davis2" %}

When I began my PhD proram in 2015, [I started working with the ONT MinION](https://twitter.com/monsterbashseq/status/963592729241382912) to see if it is feasible to use for genome assemblies of eukaryotic organisms. It turns out that it is not, at least for these killifish. To achieve 30-50x coverage for a genome assembly of a 1-1.1 Gb killifish genome, we would need to use 30-50 MinION flowcells. At a [cost of $800-900/flowcell](https://docs.google.com/spreadsheets/d/1yBncz75kgwExCXy7sC9LsMaDGs8OJJJGg9f4o3DcoQE/edit?usp=sharing), that is not a reasonably affordable genome ($24,000-45,000). Whereas it took [39 flowcells to get ~30x (90Gb) coverage of the human genome](https://www.biorxiv.org/content/early/2017/04/20/128835), one PromethION flowcells has sequenced [~80x coverage of human genome](https://gigabaseorgigabyte.wordpress.com/2018/05/24/promethion-human-genome-na19240/). PromethION promises [higher yield](https://nanoporetech.com/about-us/news/promethion-wild-high-yield-data-announcements-human-and-plant-dna) than MinION, allowing it to be more feasible for larger genomes >1 Gb. The cost of one [PromethION run at UC Davis](http://dnatech.genomecenter.ucdavis.edu/nanopore-sequencing-ont-promethion/) is $2,880. As you see below, we have not achieved this high of a yield on Promethtion, but we have at least achieved 20-30x coverage of our killifish genomes.

The avalance of data commences!

# Four species of *Fundulus* killifish

<img style="float: right;" src="https://images-na.ssl-images-amazon.com/images/I/81SpSNXaFfL.jpg" width="200px">

These species of killifish have never been sequenced before. We have a genome for the [classic laboratory and field model](https://www.sciencedirect.com/science/article/pii/S1744117X0700072X) species, [*Fundulus heteroclitus*](http://uswest.ensembl.org/Fundulus_heteroclitus/Info/Index) for understanding how organisms and populations interact with the external environment. But we do not have any genomic information from other important related freshwater species of *Fundulus*. 

#### Why killifish?

The *Fundulus heteroclitus* killifish has been well-studied because of its physiological resilience to environmental change. It has also been found to have [rapidly evolved to polluted environments](http://science.sciencemag.org/content/354/6317/1305). I am most interested in the Fundulus as a flexible model for resilience to salinity change. While *F. heteroclitus* is flexible to salinity change, some of its sister species are not.

The physiological requirement for high (or low) concentrations of salt ions in the water has affected geographic distributions of aquatic animals throughout evolutionary history and will continue to impact species worldwide in view of future changes in climate. To study the history of adaptation to different salinities and understand the mechanisms of evolution at the molecular level, we have been studying this group of related killifish from the genus *Fundulus*, which contains species that are natively adapted to different salinities. Some *Fundulus* species tolerate a range of salinities (euryhaline) while others require a more narrow salinity range (stenohaline) in either fresh or marine waters. Unique to this family of killifish is evidence that some species from the genus *Fundulus* have radiated into fresh water bodies on three separate, parallel occasions between 5-25 million years ago, and have lost their ability to tolerate higher salinities like their euryhaline relatives. We use *Fundulus* as a comparative model system for studying the physiological and genetic mechanisms that diverge between euryhaline and stenohaline freshwater species. 

{% flickr 31318668498 small "Fundulus phylogeny" %}

* Whitehead et al. 2010. [Phylogeny of killifish osmotic tolerance.](https://www.ncbi.nlm.nih.gov/pubmed/20100216)
* Whitehead et al. 2011. [Genomic mechanisms of evolved physiological plasticity in killifish distributed along an environmental salinity gradient.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3076822/)
* Brennan et al. 2015. [Reciprocal osmotic challenges reveal mechanisms of divergence in phenotypic plasticity in the killifish Fundulus heteroclitus.](http://jeb.biologists.org/content/218/8/1212)
* Brennan et al. 2018. [Integrative Population and Physiological Genomics Reveals Mechanisms of Adaptation in Killifish.](https://academic.oup.com/mbe/advance-article/doi/10.1093/molbev/msy154/5068378?guestAccessKey=89fdd3c9-8894-4a89-acac-548f2e3a06d2)

# DNA extractions

Live samples were shipped to UC Davis by [Dr. David Duvernell](https://scholar.google.com/citations?hl=en&user=QQIh3CsAAAAJ&view_op=list_works&sortby=pubdate) and [Dr. Jake Schaefer](http://ichthyology.usm.edu/). When possible (all but Fundulus catenatus), samples were extracted from fresh tissue. Fish heads were used for high molecular weight (hmw) DNA extractions.

* [Ultra-long read sequencing protocol by Josh Quick and Nick Loman](ttps://www.protocols.io/view/ultra-long-read-sequencing-protocol-for-rad004-mrxc57n) 
* [Agilent #200600 DNA extraction kit](https://www.agilent.com/cs/library/usermanuals/public/200600.pdf) found by Tony
* Tony Gil and I found that fresh is better than frozen
* With Dr. Ruta, we tried [Qiagen](https://www.qiagen.com/us/resources/resourcedetail?id=a9e6a609-4600-4b03-afbd-974318590ce5&lang=en) - p. 39 "DNA purification from tissue using the Gentra puregene Tissue Kit"
* DNA was difficult to get into solution. (show pic)
* Dr. Ruta found that adding urea to lysis buffer helped the final DNA pellet to be less brittle and go into solution easier [Wasko et al. 2003](https://www.ncbi.nlm.nih.gov/pubmed/14641478)
* Something weird with our samples. ONT has mentioned that this could a fish-specific sample issue, based on experiences with other customers. Has anyone else found this to be the case with fish hmw DNA? Brittle and breaking easily?



[Good pf gel, thanks Tony](https://twitter.com/monsterbashseq/status/852655158160904192)

# Sequencing

Here are the ONT PromethION sequencing data we have so far:

| Species |  bases called |  n reads | avg length | largest | reads N50 |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| Adinia xenica | 38,467,326,719 | 15,704,522 | 2,449 | 953,774 | 5,733, n = 1,373,426 | <img src="https://biodiversity.utexas.edu/images/easyblog_articles/3681/adinia_xenica.jpg"> |
| Fundulus nottii | 33,440,866,723 | 5,160,367 | 6,480 | 667,947 | 12,995, n=700,534 | [<img src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcT1B0oYaT8isujaT2qLP3decYTRvuiLZGvinMEEkR0algWfnrzV">](http://www.nanfa.org/ac/fundulus-notti-starhead-toopminnow.pdf) |
| F. catenatus | 40,274,806,587 | 23,701,206 | 1,699 | 590,485 | 3,439, n = 2,687,295 | <img src="https://nature.mdc.mo.gov/sites/default/files/styles/centered_full/public/media/images/2010/05/northern_studfish.jpg?itok=kLYZJ6V4"> |
| Fundulus olivaceus | 4,962,626,713 (MinION)| 740,248 | 6,704 | 973,552 | 12726, n = 117202 | <img src="https://nature.mdc.mo.gov/sites/default/files/styles/centered_full/public/media/images/2010/05/blackspotted_topminnow.jpg?itok=1V78GVZI"> |

(We're planning to do a PromethION run with *F. olivaceus* in the next week.)

There is evidence that the [consensus assembly can be improved by using the pilon/racon program with Illumina data](http://www.g3journal.org/content/early/2018/08/07/g3.118.200160) (Miller et al. 2018).

| Species |  native physiology | clade | ONT data |  Illumina | best assembly N50 | miniasm assembly size |
| --- | --- | --- | --- | --- | --- | --- | --- |
| *Fundulus catenatus* | Freshwater | 1 | TBD | TBD | TBD | 
| *Adinia xenica* | Marine  | 3 | 38 Gb | TBD | 369,038; n = 794 | 961,615,159 |
| *Fundulus olivaceus* | Freshwater | 3 | 4.9 Gb | 180 Gb | 42,823; n=7,616 (Masurca) | 1,134,160,060 |
| *Fundulus nottii* | Freshwater | 3 | 33 Gb | TBD | 2 Mb; n = 151 | 1,141,443,860 |

Other species of freshwater and brackish water killifish (*Lucania parva* and *Lucania goodei*) from Clade 2 are being sequenced by [Dr. Becky Fuller's lab at U. Illinois Urbana-Champaign](http://beckyfullerlab.weebly.com/).

Assemblies of *Fundulus olivaceus* data:

Seq data input | Tool | bases | n_contigs | average | largest | N50 | BUSCO (eukaryota) |
--- | --- | --- | --- | --- | --- | --- | --- |
| ONT | Canu | 9,804,264 | 540 | 18,156 | 365,191 | 40,681, n = 43 | 0.7% |
| ONT | Miniasm | 4,917,546 | 153 | 32,140 | 233,136 | 50,056, n = 25 | 0.0 % |
| Illumina | Megahit | 1,183,861,293 | 1038799 | 1,139 | 88,218 | 3,846, n = 77,800 | 45.6 % |
| Illumina | ABySS | 1,381,148,284 | 1024759 | 1,347 | 140629 | 9833, n = 37,013 | 77.9% |
| Hybrid | Alpaca | xxx | 
| Hybrid | Masurca | 1,134,160,060  | 90,237 | 12,568 | 386,222 | 42,823, n = 7,616 | 86.2% |
| Hybrid | SSPACE | xxx |

# Preliminary data products

Rough assemblies and QC are available on [osf](https://osf.io/zjv86/).

```
pip install osfclient
export OSF_PASSWORD=password
export OSF_USERNAME=email.address
# list files in OSF
osf -p zjv86 ls
# copy files from OSF to local (or hpc - wherever you're working)
osf -p zjv86 clone Folivaceus_hybrid_genome_assembly 
```

# References

Jain M et al. 2017. [A Fast Approximate Algorithm for Mapping Long Reads to Large Reference Databases.](https://www.biorxiv.org/content/early/2017/03/24/103812)

Loose et al. 2016. [Real time selective sequencing using nanopore technology.](https://www.biorxiv.org/content/early/2016/02/03/038760)
>The anticipated increasing speed of nanopore sequencing ('fast mode') and
the scaling up of the MinION to 3,000 channels, and the PromethION with 144,000 channels, will challenge the implementation of 'Read Until' in real time and require algorithmic enhancements and computational power

Jain C et al. 2017. [Nanopore sequencing and assembly of a human genome with ultra-long reads.](https://www.biorxiv.org/content/biorxiv/early/2017/04/20/128835.full.pdf)

[Nice slides by Benjamin Schwessinger explainig wet lab comparisons for hmw DNA prep.](https://f1000research.com/slides/7-367)

Tan et al. 2018. [Finding Nemo: hybrid assembly with Oxford Nanopore and Illumina reads greatly improves the clownfish (*Amphiprion ocellaris*) genome assembly.](https://academic.oup.com/gigascience/article/7/3/gix137/4803946)

Miller et al. 2018. [GENOME REPORT: Highly Contiguous Genome Assemblies of 15 Drosophila Species Generated Using Nanopore Sequencing.](http://www.g3journal.org/content/early/2018/08/07/g3.118.200160)

Duc Cao et al. 2016. [Scaffolding and Completing Genome Assemblies in Real-time with Nanopore Sequencing](https://www.biorxiv.org/content/early/2016/05/22/054783)

Tyler et al. 2018. [Evaluation of Oxford Nanoporeâ€™s MinION Sequencing Device for Microbial Whole Genome Sequencing Applications](https://www.nature.com/articles/s41598-018-29334-5.pdf)

De Roeck et al. 2018. [Accurate characterization of expanded tandem repeat length and sequence through whole genome long-read sequencing on PromethION.](https://www.biorxiv.org/content/early/2018/10/09/439026)

# List of available tools for ONT data:

Suggestions for additions to this list are welcome! (Especially if you have experience with them working well!)

### QC & Evaluation:
* [Porechop](https://github.com/rrwick/Porechop): removing adapters
* [NanoPlot](https://anaconda.org/bioconda/nanoplot): quality report
* [sourmash](http://sourmash.readthedocs.io/en/latest/tutorials.html): identify contamination
* [BUSCO](chrome-extension://oemmndcbldboiebfnladdacbdfmadadm/https://gitlab.com/ezlab/busco/raw/master/BUSCO_v3_userguide.pdf): content evaluation
* [Nanocomp](https://github.com/wdecoster/nanocomp): compare multiple ONT runs
* [mashmap](https://github.com/marbl/MashMap): for visualization (replaces nucmer)

### Assemblers:

* [canu, long reads](https://github.com/marbl/canu/releases): [quick start](http://canu.readthedocs.io/en/latest/quick-start.html), [tutorial](https://github.com/marbl/canu/blob/master/documentation/source/tutorial.rst) and [paper](http://genome.cshlp.org/content/27/5/722.full)
* [masurca, hybrid assembly](https://github.com/alekseyzimin/masurca) (on [conda](https://anaconda.org/bioconda/masurca))
* [SPAdes hybrid assembly](http://cab.spbu.ru/files/release3.11.1/manual.html) (for bacteria)
* [Alpaca, hybrid assembly](https://github.com/VicugnaPacos/ALPACA/) and [paper](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-017-3927-8)
* [SSPACE-LongRead scaffolding](https://www.baseclear.com/genomics/bioinformatics/basetools/SSPACE-longread)
(no instructions, but a [paper](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-15-211)!), here are [my instructions for SSPACE-LongRead](https://github.com/ljcohen/hybrid_genome_assembly/blob/master/SSPACE/README.md)

### Polishing and consensus improvement:
* [Unicycler polish](https://github.com/rrwick/Unicycler/blob/master/docs/unicycler-polish.md) ([paper])(http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005595)
* [Nanopore polish, input ONT assembly with Illumina data](https://github.com/nanoporetech/ont-assembly-polish)
* [Nanopolish](https://github.com/jts/nanopolish)
* [Nanocorr](https://github.com/jgurtowski/nanocorr): error correction (See [presentation](http://schatzlab.cshl.edu/presentations/2015/2015.02.28.AGBT%20Nanocorr%20Assembly.pdf))
* [racon](https://github.com/isovic/racon)
* [pilon](https://github.com/broadinstitute/pilon)
* [npScarf](https://github.com/mdcao/npScarf), [paper](https://www.nature.com/articles/ncomms14515)

Here is the [pipeline](https://github.com/danrdanny/Drosophila15GenomesProject/) from [Miller et al. 2018 15 Drosophila genomes paper](http://www.g3journal.org/content/ggg/8/10/3131.full.pdf).

# Example rough workflow steps:

Install miniconda Python package manager:
```
wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
bash Miniconda3-latest-Linux-x86_64.sh -b
echo export PATH="$HOME/miniconda3/bin:$PATH" >> ~/.bashrc
source ~/.bashrc
```

Create a conda environment for software installation:
```
conda create -n ONT python=3
source activate ONT
conda config --add channels defaults
conda config --add channels conda-forge
conda config --add channels bioconda
```

Install [minimap2](https://github.com/lh3/minimap2) (minimap2-2.12), [miniasm](https://github.com/lh3/miniasm) and other tools:
```
conda install assembly-stats minimap2 miniasm bwa samtools busco pilon mashmap
```

Assemble:
```
# Overlap
minimap2 -x ava-ont -t8 A_xenica_combined.fastq | gzip -1 > A_xenica_reads.paf.gz
# Layout
miniasm -f A_xenica_combined.fastq A_xenica_reads.paf.gz > reads.gfa
```

Use pilon to improve consensus:
```
bwa index <genome.fasta>
bwa mem -t 4 genome.fasta R1.fq R2.fq | samtools view -Sb - | samtools sort - -o ecto_old_paired_qc.sorted.bam
samtools index ecto_old_paired_qc.sorted.bam
pilon --genome F_olivaceus.canu.ONT.unitigs.fasta --bam F_olivaceus_ONT_polished.sorted.bam --threads 8
```

BUSCO for evaluation:
```
run_BUSCO.py -i F_nottii_miniasm_ONT.fa -l ${BUSCO_LINEAGE}/eukaryota_odb9 -o F_nottii_miniasm --mode genome -c 8
```

Visualize, run [mashmap](https://github.com/marbl/MashMap):
```
mashmap \
-r /mnt/home/ljcohen/reference/GCF_000826765.1_Fundulus_heteroclitus-3.0.2_genomic.fna \
-q ../../masurca_Folivaceus/CA.mr.41.15.15.0.02/final.genome.scf.fasta
```

Plot:
```
generateDotPlot png large mashmap.out
```
Edit the `out.gp` file, because of reasons.
```
vi out.gp
```
* Delete word 'tiny' in first line
* Comment out 3 lines with word 'mouse' in them.
Then run:
```
gnuplot out.gp
```
which will then produce the png file.
