Title: Killifish PromethION data sharing
Date: 2018-10-08
Category: Python
Tags: sequencing, killifish, Oxford Nanopore Technologies
Author: Lisa K. Johnson
Summary: Sharing PromethION sequencing data from 4 species of Fundulus killifish

We have some new long-read [ONT PromethION]() genomic DNA sequencing data from killifish! While we're working with the data and preparing for publication, we wanted to share this information and data in case it is useful for anyone. Since there are not many PromethION instruments available, these are probably some of the first data to be shared. At least from killifish.

This is a collaborative effort between my two advisors, Dr. Titus Brown and Dr. Andrew Whitehead. (Seriously, a grad student could not be luckier than I am to have two great advisors and collaborative support from great labs and colleagues.) Other contributors are Dr. Ruta and Tony Gill. Special thanks to PoreCamp, Texas A&M Agrilife Sequencing. And Dr. David Duvernell for sending us fish. And thank you to supported provided by the Moore Foundation Data-Driven-Discovery investigator award to Dr. Titus Brown.

# New PromethION instrument at the UC Davis Genome Center
In [May 2018](https://twitter.com/meganamsu/status/997216050247864322), the [UC Davis Genome Center](http://dnatech.genomecenter.ucdavis.edu/nanopore-sequencing-ont-promethion/) received its PromethION from Oxford Nanopore. 

This was part of a collaborative effort to acquire this instrument through the early release program, back in 2016 by Genome Center faculty: Dr. Richard Michelmore, Dr. Megan Dennis, and Dr. Titus Brown.

{% flickr 40821422010 [small] ["PromethION at UC Davis"] %}

{% flickr 42629095711 [small] ["PromethION at UC Davis2"] %}

I have been working with the ONT MinION to see if it is feasible to use for genome assemblies of eukaryotic organisms. 

https://twitter.com/monsterbashseq/status/963592729241382912

* MinION Costs: https://twitter.com/monsterbashseq/status/963154668942589952

* I went to Porecamp: https://twitter.com/monsterbashseq/status/959153636445073408

* Turns out that it is not. But the PromethION has been great for our purposes.
* PromethION promises [higher yield](https://nanoporetech.com/about-us/news/promethion-wild-high-yield-data-announcements-human-and-plant-dna) than MinION, allowing it to be more feasible for 

# *Fundulus* killifish

<img align="right" height="300" src="https://images-na.ssl-images-amazon.com/images/I/81SpSNXaFfL.jpg">

These species of killifish have never been sequenced before. Live samples were gratiously shipped to UC Davis by [Dr. David Duvernell at Missouri University of Science & Technology](http://people.mst.edu/faculty/duvernelld/index.html)

{% flickr 24012654907 [small] ["shipping container"] %}

{% flickr 25007459318 [small] ["breathe bag"] %}

https://twitter.com/monsterbashseq/status/1048308106596245504

Very important fish for comparative genomics:

{% flickr 31318668498 [medium] ["Fundulus 3 clades"] %}

* [Whitehead et al. 2010. Phylogeny of killifish osmotic tolerance](https://www.ncbi.nlm.nih.gov/pubmed/20100216)
* [Whitehead et al. 2011. Genomic mechanisms of evolved physiological plasticity in killifish distributed along an environmental salinity gradient](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3076822/)
* [Brennan et al. 2015. Reciprocal osmotic challenges reveal mechanisms of divergence in phenotypic plasticity in the killifish Fundulus heteroclitus](http://jeb.biologists.org/content/218/8/1212)
* Recently, [Fundulidae tree of life](https://fishtreeoflife.org/taxonomy/family/Fundulidae/)

# DNA extractions

* long-reads protocol by Joshua Quick and Nick Loman: https://twitter.com/Scalene/status/955091398285582337
* Agilent protocol, found by Tony
* Fresh is better than frozen
* Tried other protocols
* Ruta found that adding urea to lysis buffer helped
* DNA was difficult to get into solution.
* With the MinION, our sample was not utilizing the pores as efficiently as it could be. ONT has mentioned that this is a fish-specific sample issue. Has anyone else found this to be the case with fish hmw DNA? Brittle and breaking easily. 
* Show gels.

https://twitter.com/monsterbashseq/status/852655158160904192

https://twitter.com/nanopore/status/872910510508953605

# Sequencing

Species |  native physiology | clade | ONT data |  Illumina | best N50 | |
--- | --- | --- | --- | --- | --- | --- | --- | 
*Fundulus catenatus* | Freshwater | 1 | TBD | TBD | TBD | <img src="https://nature.mdc.mo.gov/sites/default/files/styles/centered_full/public/media/images/2010/05/northern_studfish.jpg?itok=kLYZJ6V4">
*Adinia xenica* | Marine  | 3 | 38 Gb | TBD | 369,038; n = 794 | <img src="https://biodiversity.utexas.edu/images/easyblog_articles/3681/adinia_xenica.jpg"> 
*Fundulus olivaceus* | Freshwater | 3 | 4.9 Gb | 180 Gb | 42,823; n=7,616 (Masurca) | <img src="https://nature.mdc.mo.gov/sites/default/files/styles/centered_full/public/media/images/2010/05/blackspotted_topminnow.jpg?itok=1V78GVZI">
*Fundulus nottii* | Freshwater | 3 | 33 Gb | TBD | 2 Mb; n = 151 | [<img src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcT1B0oYaT8isujaT2qLP3decYTRvuiLZGvinMEEkR0algWfnrzV">](http://www.nanfa.org/ac/fundulus-notti-starhead-toopminnow.pdf)

Other species of freshwater killifish (*Lucania parva* and *Lucania goodei*) are being sequenced by [Dr. Becky Fuller's lab at U. Illinois Urbana-Champaign](http://beckyfullerlab.weebly.com/).


## Preliminary data products

Rough assemblies and QC:

https://osf.io/zjv86/


# Our plan from here:

## 1. Post data

* [European Nucleotide Archive (ENA)](https://www.ebi.ac.uk/ena/submit/sequence-submission) 
* [zenodo has recently upgraded their file size limits to 50 GB](http://help.zenodo.org/), however they require a request, which I've submitted 3 and no response in 12 days.
* NCBI - is ridiculous

## 2. Data note

Either:
- [F1000 research](https://f1000research.com/for-authors/article-guidelines/data-notes)
- [GigaScience](https://academic.oup.com/gigascience/pages/data_note)

## 3. Science!
Comparative genomics. Scan for regulatory regions of interest, e.g. [salinity-responsive enhancing elements (OSRE)](http://www.pnas.org/content/114/13/E2729)

# Questions

* What to do with millions of fast5 files? Our fastq are 70-100 GB, but the directory of raw fast5 signal files are >600 GB. [Nanopolish](https://github.com/jts/nanopolish) uses the raw fast5 signal files to correct. How can these be shared efficiently?
* Is sharing the data useful for anyone? 
* Any advice, questions, or comments?


