Title: Explorations with Orthofinder
Date: 2018-12-05
Category: Orthologues
Tags: comaparative
Author: Lisa K. Johnson
Summary: Running Orthofinder for the first time, taking a look at the output files

Here's my weekend exploration of [Orthofinder](https://github.com/davidemms/OrthoFinder) with [amino acid translations from de novo transcriptome assemblies of 17 species of killifish](http://farm.cse.ucdavis.edu/~ljcohen/killifish/aa_translations/). Translations of Trinity de novo transcirptomes were done with transdecoder by the [dammit annotation pipeline](https://github.com/dib-lab/dammit).

[Orthofinder manual](https://github.com/davidemms/OrthoFinder/blob/master/OrthoFinder-manual.pdf) and a preprint on [biorxiv](https://www.biorxiv.org/content/early/2018/11/08/466201). Found this via a [tweet](https://twitter.com/David__Emms/status/1060808772996743168), and [tweet recommendation](https://twitter.com/PhilippBayer/status/1060821270860718082).

Here's the script that I ran, submitted to the barbera cluster at UC Davis (which took ~1 week to get through the queue over Thanksgiving break):
```
johnsolk@barbera:/share/pilot-johnsolk$ cat orthofinder.sbatch 
#!/bin/bash -l
#SBATCH -J orthofinder_kfish
#SBATCH -t 72:00:00
#SBATCH -c 8
#SBATCH -p production
#SBATCH --mem=48000

aklog
source ~/.bashrc
source activate py27
orthofinder -f killifish_RNAseq_fasta -S diamond
```
Output files:
```
johnsolk@barbera:/share/pilot-johnsolk/killifish_RNAseq_fasta/Results_Nov26$ ls -lah
total 388M
drwxrwxr-x 4 johnsolk johnsolk   12 Nov 27 16:50 .
drwxrwxr-x 5 johnsolk johnsolk   22 Nov 26 11:23 ..
-rw-rw-r-- 1 johnsolk johnsolk 411M Nov 27 16:49 Orthogroups.csv
-rw-rw-r-- 1 johnsolk johnsolk 6.1M Nov 27 16:49 Orthogroups.GeneCount.csv
-rw-rw-r-- 1 johnsolk johnsolk 3.6K Nov 27 16:50 Orthogroups_SpeciesOverlaps.csv
-rw-rw-r-- 1 johnsolk johnsolk 449M Nov 27 16:48 Orthogroups.txt
-rw-rw-r-- 1 johnsolk johnsolk  45M Nov 27 16:49 Orthogroups_UnassignedGenes.csv
drwxrwxr-x 6 johnsolk johnsolk   14 Nov 28 10:35 Orthologues_Nov27
-rw-rw-r-- 1 johnsolk johnsolk   69 Nov 27 16:50 SingleCopyOrthogroups.txt
-rw-rw-r-- 1 johnsolk johnsolk 1.7K Nov 27 16:50 Statistics_Overall.csv
-rw-rw-r-- 1 johnsolk johnsolk 9.7K Nov 27 16:50 Statistics_PerSpecies.csv
drwxrwxr-x 2 johnsolk johnsolk  330 Nov 27 16:47 WorkingDirectory
```
[Orthofinder workflow](https://github.com/davidemms/OrthoFinder):
![](https://raw.githubusercontent.com/davidemms/OrthoFinder/master/orthofinder/workflow.png)

Orthologues directory:
```
johnsolk@barbera:/share/pilot-johnsolk/killifish_RNAseq_fasta/Results_Nov26/Orthologues_Nov27$ ls -lah
total 462M
drwxrwxr-x  6 johnsolk johnsolk   14 Nov 28 10:35 .
drwxrwxr-x  4 johnsolk johnsolk   12 Nov 27 16:50 ..
-rw-rw-r--  1 johnsolk johnsolk 2.2G Nov 28 10:35 Duplications.csv
drwxrwxr-x  2 johnsolk johnsolk  93K Nov 28 03:09 Gene_Trees
drwxrwxr-x 20 johnsolk johnsolk   20 Nov 28 03:10 Orthologues
-rw-rw-r--  1 johnsolk johnsolk 4.1K Nov 28 10:35 OrthologuesStats_many-to-many.csv
-rw-rw-r--  1 johnsolk johnsolk 4.1K Nov 28 10:35 OrthologuesStats_many-to-one.csv
-rw-rw-r--  1 johnsolk johnsolk 3.9K Nov 28 10:35 OrthologuesStats_one-to-many.csv
-rw-rw-r--  1 johnsolk johnsolk 4.1K Nov 28 10:35 OrthologuesStats_one-to-one.csv
-rw-rw-r--  1 johnsolk johnsolk 4.1K Nov 28 10:35 OrthologuesStats_Totals.csv
drwxrwxr-x  2 johnsolk johnsolk  93K Nov 28 10:35 Recon_Gene_Trees
-rw-rw-r--  1 johnsolk johnsolk 1.3K Nov 28 03:10 SpeciesTree_rooted_node_labels.txt
-rw-rw-r--  1 johnsolk johnsolk 1.4K Nov 28 03:10 SpeciesTree_rooted.txt
drwxrwxr-x  5 johnsolk johnsolk    8 Nov 28 10:37 WorkingDirectory
```

### Trees

* Files are in [newick format](http://evolution.genetics.washington.edu/phylip/newicktree.html) ([Wiki](https://en.wikipedia.org/wiki/Newick_format)), install [Dendroscope](http://dendroscope.org/) to view.

### Orthologue output

Orthologues are located here, these are directories (one for each species):
```
johnsolk@barbera:/share/pilot-johnsolk/killifish_RNAseq_fasta/Results_Nov26/Orthologues_Nov27/Orthologues$ ls -lah
total 1.4M
drwxrwxr-x 20 johnsolk johnsolk 20 Nov 28 03:10 .
drwxrwxr-x  6 johnsolk johnsolk 14 Nov 28 10:35 ..
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:11 Orthologues_A_xenica.trinity_out.Trinity.fasta.transdecoder.pep
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:11 Orthologues_F_catanatus.trinity_out.Trinity.fasta.transdecoder.pep
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:11 Orthologues_F_chrysotus.trinity_out.Trinity.fasta.transdecoder.pep
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:11 Orthologues_F_diaphanus.trinity_out.Trinity.fasta.transdecoder.pep
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:11 Orthologues_F_grandis.trinity_out.Trinity.fasta.transdecoder.pep
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:11 Orthologues_F_heteroclitusMDPL.trinity_out.Trinity.fasta.transdecoder.pep
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:11 Orthologues_F_heteroclitusMDPP.trinity_out.Trinity.fasta.transdecoder.pep
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:11 Orthologues_F_notatus.trinity_out.Trinity.fasta.transdecoder.pep
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:11 Orthologues_F_notti.trinity_out.Trinity.fasta.transdecoder.pep
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:11 Orthologues_F_olivaceous.trinity_out.Trinity.fasta.transdecoder.pep
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:12 Orthologues_F_parvapinis.trinity_out.Trinity.fasta.transdecoder.pep
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:11 Orthologues_F_rathbuni.trinity_out.Trinity.fasta.transdecoder.pep
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:11 Orthologues_F_sciadicus.trinity_out.Trinity.fasta.transdecoder.pep
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:11 Orthologues_F_similis.trinity_out.Trinity.fasta.transdecoder.pep
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:11 Orthologues_F_zebrinus.trinity_out.Trinity.fasta.transdecoder.pep
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:11 Orthologues_L_goodei.trinity_out.Trinity.fasta.transdecoder.pep
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:11 Orthologues_L_parva.trinity_out.Trinity.fasta.transdecoder.pep
drwxrwxr-x  2 johnsolk johnsolk 19 Nov 28 03:10 Putative_Xenologues
johnsolk@barbera:/share/pilot-johnsolk/killifish_RNAseq_fasta/Results_Nov26/Orthologues_Nov27/Orthologues$ pwd
/share/pilot-johnsolk/killifish_RNAseq_fasta/Results_Nov26/Orthologues_Nov27/Orthologues
```
These are for *A. xenica*:
```
johnsolk@barbera:/share/pilot-johnsolk/killifish_RNAseq_fasta/Results_Nov26/Orthologues_Nov27/Orthologues/Orthologues_A_xenica.trinity_out.Trinity.fasta.transdecoder.pep$ ls -lah
total 180M
drwxrwxr-x  2 johnsolk johnsolk   19 Nov 28 03:11 .
drwxrwxr-x 20 johnsolk johnsolk   20 Nov 28 03:10 ..
-rw-rw-r--  1 johnsolk johnsolk  27M Nov 28 10:35 A_xenica.trinity_out.Trinity.fasta.transdecoder.pep__v__F_catanatus.trinity_out.Trinity.fasta.transdecoder.pep.csv
-rw-rw-r--  1 johnsolk johnsolk  27M Nov 28 10:35 A_xenica.trinity_out.Trinity.fasta.transdecoder.pep__v__F_chrysotus.trinity_out.Trinity.fasta.transdecoder.pep.csv
-rw-rw-r--  1 johnsolk johnsolk  24M Nov 28 10:35 A_xenica.trinity_out.Trinity.fasta.transdecoder.pep__v__F_diaphanus.trinity_out.Trinity.fasta.transdecoder.pep.csv
-rw-rw-r--  1 johnsolk johnsolk  29M Nov 28 10:35 A_xenica.trinity_out.Trinity.fasta.transdecoder.pep__v__F_grandis.trinity_out.Trinity.fasta.transdecoder.pep.csv
-rw-rw-r--  1 johnsolk johnsolk  26M Nov 28 10:35 A_xenica.trinity_out.Trinity.fasta.transdecoder.pep__v__F_heteroclitusMDPL.trinity_out.Trinity.fasta.transdecoder.pep.csv
-rw-rw-r--  1 johnsolk johnsolk  26M Nov 28 10:35 A_xenica.trinity_out.Trinity.fasta.transdecoder.pep__v__F_heteroclitusMDPP.trinity_out.Trinity.fasta.transdecoder.pep.csv
-rw-rw-r--  1 johnsolk johnsolk  28M Nov 28 10:35 A_xenica.trinity_out.Trinity.fasta.transdecoder.pep__v__F_notatus.trinity_out.Trinity.fasta.transdecoder.pep.csv
-rw-rw-r--  1 johnsolk johnsolk  17M Nov 28 10:35 A_xenica.trinity_out.Trinity.fasta.transdecoder.pep__v__F_notti.trinity_out.Trinity.fasta.transdecoder.pep.csv
-rw-rw-r--  1 johnsolk johnsolk  26M Nov 28 10:35 A_xenica.trinity_out.Trinity.fasta.transdecoder.pep__v__F_olivaceous.trinity_out.Trinity.fasta.transdecoder.pep.csv
-rw-rw-r--  1 johnsolk johnsolk  24M Nov 28 10:35 A_xenica.trinity_out.Trinity.fasta.transdecoder.pep__v__F_parvapinis.trinity_out.Trinity.fasta.transdecoder.pep.csv
-rw-rw-r--  1 johnsolk johnsolk  29M Nov 28 10:35 A_xenica.trinity_out.Trinity.fasta.transdecoder.pep__v__F_rathbuni.trinity_out.Trinity.fasta.transdecoder.pep.csv
-rw-rw-r--  1 johnsolk johnsolk  22M Nov 28 10:35 A_xenica.trinity_out.Trinity.fasta.transdecoder.pep__v__F_sciadicus.trinity_out.Trinity.fasta.transdecoder.pep.csv
-rw-rw-r--  1 johnsolk johnsolk  24M Nov 28 10:35 A_xenica.trinity_out.Trinity.fasta.transdecoder.pep__v__F_similis.trinity_out.Trinity.fasta.transdecoder.pep.csv
-rw-rw-r--  1 johnsolk johnsolk  21M Nov 28 10:35 A_xenica.trinity_out.Trinity.fasta.transdecoder.pep__v__F_zebrinus.trinity_out.Trinity.fasta.transdecoder.pep.csv
-rw-rw-r--  1 johnsolk johnsolk  25M Nov 28 10:35 A_xenica.trinity_out.Trinity.fasta.transdecoder.pep__v__L_goodei.trinity_out.Trinity.fasta.transdecoder.pep.csv
-rw-rw-r--  1 johnsolk johnsolk  27M Nov 28 10:35 A_xenica.trinity_out.Trinity.fasta.transdecoder.pep__v__L_parva.trinity_out.Trinity.fasta.transdecoder.pep.csv
-rw-rw-r--  1 johnsolk johnsolk 220K Nov 28 08:33 Putative_Horizontal_Gene_Transfer.txt
```

Output:
```
johnsolk@barbera:/share/pilot-johnsolk$ cat slurm-20077670.out
Auks API request failed : krb5 cred : unable to read credential cache
/bin/bash: /home/johnsolk/.bash_profile: Permission denied
aklog: Couldn't determine realm of user:aklog: No credentials cache found  while getting realm
/var/spool/slurmd/job20077670/slurm_script: line 9: /home/johnsolk/.bashrc: Permission denied

OrthoFinder version 2.2.7 Copyright (C) 2014 David Emms

2018-11-26 11:23:26 : Starting OrthoFinder
16 thread(s) for highly parallel tasks (BLAST searches etc.)
1 thread(s) for OrthoFinder algorithm

Checking required programs are installed
----------------------------------------
Test can run "mcl -h" - ok
Test can run "fastme -i /share/pilot-johnsolk/killifish_RNAseq_fasta/SimpleTest.phy -o /share/pilot-johnsolk/killifish_RNAseq_fasta/SimpleTest.tre" - ok

Dividing up work for BLAST for parallel processing
--------------------------------------------------
2018-11-26 11:24:53 : Creating diamond database 1 of 17
2018-11-26 11:24:56 : Creating diamond database 2 of 17
2018-11-26 11:24:59 : Creating diamond database 3 of 17
2018-11-26 11:25:01 : Creating diamond database 4 of 17
2018-11-26 11:25:03 : Creating diamond database 5 of 17
2018-11-26 11:25:06 : Creating diamond database 6 of 17
2018-11-26 11:25:09 : Creating diamond database 7 of 17
2018-11-26 11:25:11 : Creating diamond database 8 of 17
2018-11-26 11:25:14 : Creating diamond database 9 of 17
2018-11-26 11:25:15 : Creating diamond database 10 of 17
2018-11-26 11:25:18 : Creating diamond database 11 of 17
2018-11-26 11:25:20 : Creating diamond database 12 of 17
2018-11-26 11:25:24 : Creating diamond database 13 of 17
2018-11-26 11:25:26 : Creating diamond database 14 of 17
2018-11-26 11:25:28 : Creating diamond database 15 of 17
2018-11-26 11:25:30 : Creating diamond database 16 of 17
2018-11-26 11:25:32 : Creating diamond database 17 of 17

Running diamond all-versus-all
------------------------------
Using 16 thread(s)
2018-11-26 11:25:35 : This may take some time....
2018-11-27 01:40:54 : Done 200 of 289
2018-11-26 19:56:13 : Done 100 of 289
2018-11-26 11:25:35 : Done 0 of 289
2018-11-27 05:34:15 : Done all-versus-all sequence search

Running OrthoFinder algorithm
-----------------------------
2018-11-27 05:34:33 : Initial processing of each species
2018-11-27 05:56:01 : Initial processing of species 0 complete
2018-11-27 06:19:34 : Initial processing of species 1 complete
2018-11-27 06:42:17 : Initial processing of species 2 complete
2018-11-27 07:00:28 : Initial processing of species 3 complete
2018-11-27 07:26:45 : Initial processing of species 4 complete
2018-11-27 07:48:38 : Initial processing of species 5 complete
2018-11-27 08:10:19 : Initial processing of species 6 complete
2018-11-27 08:34:54 : Initial processing of species 7 complete
2018-11-27 08:45:45 : Initial processing of species 8 complete
2018-11-27 09:07:35 : Initial processing of species 9 complete
2018-11-27 09:27:08 : Initial processing of species 10 complete
2018-11-27 09:52:50 : Initial processing of species 11 complete
2018-11-27 10:08:33 : Initial processing of species 12 complete
2018-11-27 10:27:23 : Initial processing of species 13 complete
2018-11-27 10:42:00 : Initial processing of species 14 complete
2018-11-27 11:00:31 : Initial processing of species 15 complete
2018-11-27 11:25:14 : Initial processing of species 16 complete
2018-11-27 11:59:32 : Connected putatitive homologs
[mcl] cut <61> instances of overlap
2018-11-27 16:47:54 : Ran MCL

Writing orthogroups to file
---------------------------
A duplicate accession was found using just first part: Transcript_74|m.19
Tried to use only the first part of the accession in order to list the sequences in each orthogroup
more concisely but these were not unique. The full accession line will be used instead.

Orthogroups have been written to tab-delimited files:
   /share/pilot-johnsolk/killifish_RNAseq_fasta/Results_Nov26/Orthogroups.csv
   /share/pilot-johnsolk/killifish_RNAseq_fasta/Results_Nov26/Orthogroups.txt (OrthoMCL format)
   /share/pilot-johnsolk/killifish_RNAseq_fasta/Results_Nov26/Orthogroups_UnassignedGenes.csv
2018-11-27 16:50:24 : Done orthogroups

Analysing Orthogroups
=====================

Calculating gene distances
--------------------------
2018-11-27 23:24:44 : Done 0 of 95230
2018-11-27 23:36:14 : Done 1000 of 95230
2018-11-27 23:38:41 : Done 2000 of 95230
2018-11-27 23:40:18 : Done 3000 of 95230
2018-11-27 23:41:34 : Done 4000 of 95230
2018-11-27 23:42:36 : Done 5000 of 95230
2018-11-27 23:43:31 : Done 6000 of 95230
2018-11-27 23:44:19 : Done 7000 of 95230
2018-11-27 23:45:03 : Done 8000 of 95230
2018-11-27 23:45:44 : Done 9000 of 95230
2018-11-27 23:46:21 : Done 10000 of 95230
2018-11-27 23:46:58 : Done 11000 of 95230
2018-11-27 23:47:32 : Done 12000 of 95230
2018-11-27 23:48:04 : Done 13000 of 95230
2018-11-27 23:48:32 : Done 14000 of 95230
2018-11-27 23:49:01 : Done 15000 of 95230
2018-11-27 23:49:27 : Done 16000 of 95230
2018-11-27 23:49:53 : Done 17000 of 95230
2018-11-27 23:50:18 : Done 18000 of 95230
2018-11-27 23:50:43 : Done 19000 of 95230
2018-11-27 23:51:07 : Done 20000 of 95230
2018-11-27 23:51:29 : Done 21000 of 95230
2018-11-27 23:51:49 : Done 22000 of 95230
2018-11-27 23:52:09 : Done 23000 of 95230
2018-11-27 23:52:28 : Done 24000 of 95230
2018-11-27 23:52:47 : Done 25000 of 95230
2018-11-27 23:53:07 : Done 26000 of 95230
2018-11-27 23:53:26 : Done 27000 of 95230
2018-11-27 23:53:44 : Done 28000 of 95230
2018-11-27 23:54:01 : Done 29000 of 95230
2018-11-27 23:54:19 : Done 30000 of 95230
2018-11-27 23:54:37 : Done 31000 of 95230
2018-11-27 23:54:54 : Done 32000 of 95230
2018-11-27 23:55:14 : Done 33000 of 95230
2018-11-27 23:55:32 : Done 34000 of 95230
2018-11-27 23:55:50 : Done 35000 of 95230
2018-11-27 23:56:06 : Done 36000 of 95230
2018-11-27 23:56:22 : Done 37000 of 95230
2018-11-27 23:56:37 : Done 38000 of 95230
2018-11-27 23:56:53 : Done 39000 of 95230
2018-11-27 23:57:09 : Done 40000 of 95230
2018-11-27 23:57:25 : Done 41000 of 95230
2018-11-27 23:57:41 : Done 42000 of 95230
2018-11-27 23:57:58 : Done 43000 of 95230
2018-11-27 23:58:15 : Done 44000 of 95230
2018-11-27 23:58:31 : Done 45000 of 95230
2018-11-27 23:58:47 : Done 46000 of 95230
2018-11-27 23:59:02 : Done 47000 of 95230
2018-11-27 23:59:18 : Done 48000 of 95230
2018-11-27 23:59:35 : Done 49000 of 95230
2018-11-27 23:59:51 : Done 50000 of 95230
2018-11-28 00:00:06 : Done 51000 of 95230
2018-11-28 00:00:21 : Done 52000 of 95230
2018-11-28 00:00:37 : Done 53000 of 95230
2018-11-28 00:00:52 : Done 54000 of 95230
2018-11-28 00:01:07 : Done 55000 of 95230
2018-11-28 00:01:23 : Done 56000 of 95230
2018-11-28 00:01:40 : Done 57000 of 95230
2018-11-28 00:01:56 : Done 58000 of 95230
2018-11-28 00:02:11 : Done 59000 of 95230
2018-11-28 00:02:29 : Done 60000 of 95230
2018-11-28 00:02:47 : Done 61000 of 95230
2018-11-28 00:03:07 : Done 62000 of 95230
2018-11-28 00:03:27 : Done 63000 of 95230
2018-11-28 00:03:48 : Done 64000 of 95230
2018-11-28 00:04:09 : Done 65000 of 95230
2018-11-28 00:04:31 : Done 66000 of 95230
2018-11-28 00:04:52 : Done 67000 of 95230
2018-11-28 00:05:13 : Done 68000 of 95230
2018-11-28 00:05:35 : Done 69000 of 95230
2018-11-28 00:05:55 : Done 70000 of 95230
2018-11-28 00:06:14 : Done 71000 of 95230
2018-11-28 00:06:34 : Done 72000 of 95230
2018-11-28 00:06:55 : Done 73000 of 95230
2018-11-28 00:07:16 : Done 74000 of 95230
2018-11-28 00:07:36 : Done 75000 of 95230
2018-11-28 00:07:56 : Done 76000 of 95230
2018-11-28 00:08:18 : Done 77000 of 95230
2018-11-28 00:08:39 : Done 78000 of 95230
2018-11-28 00:09:02 : Done 79000 of 95230
2018-11-28 00:09:22 : Done 80000 of 95230
2018-11-28 00:09:42 : Done 81000 of 95230
2018-11-28 00:10:03 : Done 82000 of 95230
2018-11-28 00:10:24 : Done 83000 of 95230
2018-11-28 00:10:47 : Done 84000 of 95230
2018-11-28 00:11:09 : Done 85000 of 95230
2018-11-28 00:11:29 : Done 86000 of 95230
2018-11-28 00:11:51 : Done 87000 of 95230
2018-11-28 00:12:13 : Done 88000 of 95230
2018-11-28 00:12:33 : Done 89000 of 95230
2018-11-28 00:12:55 : Done 90000 of 95230
2018-11-28 00:13:16 : Done 91000 of 95230
2018-11-28 00:13:37 : Done 92000 of 95230
2018-11-28 00:13:57 : Done 93000 of 95230
2018-11-28 00:14:19 : Done 94000 of 95230
2018-11-28 00:14:40 : Done 95000 of 95230
2018-11-27 23:24:40 : Done

Inferring gene and species trees
--------------------------------

15603 trees had all species present and will be used by STAG to infer the species tree

A duplicate accession was found using just first part: Transcript_74|m.19
Tried to use only the first part of the accession in order to list the sequences in each orthogroup
more concisely but these were not unique. The full accession line will be used instead.


Best outgroup(s) for species tree
---------------------------------
2018-11-28 03:10:02 : Starting STRIDE
2018-11-28 03:10:53 : Done STRIDE
Observed 863 well-supported, non-terminal duplications. 852 support the best root and 11 contradict it.
Best outgroup for species tree:
  L_goodei.trinity_out.Trinity.fasta.transdecoder.pep, L_parva.trinity_out.Trinity.fasta.transdecoder.pep

Reconciling gene trees and species tree
---------------------------------------
Outgroup: L_goodei.trinity_out.Trinity.fasta.transdecoder.pep, L_parva.trinity_out.Trinity.fasta.transdecoder.pep
2018-11-28 03:10:53 : Starting Recon and orthologues
2018-11-28 03:10:53 : Starting OF Orthologues
2018-11-28 03:11:01 : Done 0 of 95230
2018-11-28 03:29:22 : Done 1000 of 95230
2018-11-28 03:41:32 : Done 2000 of 95230
2018-11-28 03:52:16 : Done 3000 of 95230
2018-11-28 04:02:05 : Done 4000 of 95230
2018-11-28 04:11:43 : Done 5000 of 95230
2018-11-28 04:21:05 : Done 6000 of 95230
2018-11-28 04:29:57 : Done 7000 of 95230
2018-11-28 04:38:22 : Done 8000 of 95230
2018-11-28 04:46:36 : Done 9000 of 95230
2018-11-28 04:54:32 : Done 10000 of 95230
2018-11-28 05:02:18 : Done 11000 of 95230
2018-11-28 05:09:55 : Done 12000 of 95230
2018-11-28 05:17:17 : Done 13000 of 95230
2018-11-28 05:24:32 : Done 14000 of 95230
2018-11-28 05:31:34 : Done 15000 of 95230
2018-11-28 05:38:29 : Done 16000 of 95230
2018-11-28 05:45:15 : Done 17000 of 95230
2018-11-28 05:52:01 : Done 18000 of 95230
2018-11-28 05:58:46 : Done 19000 of 95230
2018-11-28 06:05:27 : Done 20000 of 95230
2018-11-28 06:11:59 : Done 21000 of 95230
2018-11-28 06:17:55 : Done 22000 of 95230
2018-11-28 06:24:06 : Done 23000 of 95230
2018-11-28 06:29:51 : Done 24000 of 95230
2018-11-28 06:35:17 : Done 25000 of 95230
2018-11-28 06:40:36 : Done 26000 of 95230
2018-11-28 06:45:52 : Done 27000 of 95230
2018-11-28 06:50:55 : Done 28000 of 95230
2018-11-28 06:55:59 : Done 29000 of 95230
2018-11-28 07:00:50 : Done 30000 of 95230
2018-11-28 07:05:44 : Done 31000 of 95230
2018-11-28 07:10:23 : Done 32000 of 95230
2018-11-28 07:14:57 : Done 33000 of 95230
2018-11-28 07:19:32 : Done 34000 of 95230
2018-11-28 07:24:08 : Done 35000 of 95230
2018-11-28 07:28:38 : Done 36000 of 95230
2018-11-28 07:32:55 : Done 37000 of 95230
2018-11-28 07:37:05 : Done 38000 of 95230
2018-11-28 07:41:05 : Done 39000 of 95230
2018-11-28 07:45:02 : Done 40000 of 95230
2018-11-28 07:48:54 : Done 41000 of 95230
2018-11-28 07:52:39 : Done 42000 of 95230
2018-11-28 07:56:18 : Done 43000 of 95230
2018-11-28 08:00:01 : Done 44000 of 95230
2018-11-28 08:03:32 : Done 45000 of 95230
2018-11-28 08:07:08 : Done 46000 of 95230
2018-11-28 08:10:37 : Done 47000 of 95230
2018-11-28 08:14:11 : Done 48000 of 95230
2018-11-28 08:17:40 : Done 49000 of 95230
2018-11-28 08:21:08 : Done 50000 of 95230
2018-11-28 08:24:28 : Done 51000 of 95230
2018-11-28 08:27:52 : Done 52000 of 95230
2018-11-28 08:31:12 : Done 53000 of 95230
2018-11-28 08:34:27 : Done 54000 of 95230
2018-11-28 08:37:47 : Done 55000 of 95230
2018-11-28 08:40:57 : Done 56000 of 95230
2018-11-28 08:44:03 : Done 57000 of 95230
2018-11-28 08:47:14 : Done 58000 of 95230
2018-11-28 08:50:24 : Done 59000 of 95230
2018-11-28 08:53:25 : Done 60000 of 95230
2018-11-28 08:56:33 : Done 61000 of 95230
2018-11-28 08:59:42 : Done 62000 of 95230
2018-11-28 09:02:51 : Done 63000 of 95230
2018-11-28 09:05:54 : Done 64000 of 95230
2018-11-28 09:08:57 : Done 65000 of 95230
2018-11-28 09:12:00 : Done 66000 of 95230
2018-11-28 09:15:01 : Done 67000 of 95230
2018-11-28 09:18:00 : Done 68000 of 95230
2018-11-28 09:20:56 : Done 69000 of 95230
2018-11-28 09:23:48 : Done 70000 of 95230
2018-11-28 09:26:43 : Done 71000 of 95230
2018-11-28 09:29:36 : Done 72000 of 95230
2018-11-28 09:32:30 : Done 73000 of 95230
2018-11-28 09:35:23 : Done 74000 of 95230
2018-11-28 09:38:15 : Done 75000 of 95230
2018-11-28 09:41:06 : Done 76000 of 95230
2018-11-28 09:44:01 : Done 77000 of 95230
2018-11-28 09:46:57 : Done 78000 of 95230
2018-11-28 09:49:53 : Done 79000 of 95230
2018-11-28 09:52:48 : Done 80000 of 95230
2018-11-28 09:55:42 : Done 81000 of 95230
2018-11-28 09:58:35 : Done 82000 of 95230
2018-11-28 10:01:28 : Done 83000 of 95230
2018-11-28 10:04:15 : Done 84000 of 95230
2018-11-28 10:07:00 : Done 85000 of 95230
2018-11-28 10:09:47 : Done 86000 of 95230
2018-11-28 10:12:32 : Done 87000 of 95230
2018-11-28 10:15:18 : Done 88000 of 95230
2018-11-28 10:18:09 : Done 89000 of 95230
2018-11-28 10:20:57 : Done 90000 of 95230
2018-11-28 10:23:45 : Done 91000 of 95230
2018-11-28 10:26:34 : Done 92000 of 95230
2018-11-28 10:29:21 : Done 93000 of 95230
2018-11-28 10:32:06 : Done 94000 of 95230
2018-11-28 10:34:49 : Done 95000 of 95230
2018-11-28 10:35:27 : Done OF Orthologues
2018-11-28 10:35:50 : Done Recon

Writing results files
=====================
2018-11-28 10:37:28 : Done orthologues
Orthogroups have been written to tab-delimited files:
   /share/pilot-johnsolk/killifish_RNAseq_fasta/Results_Nov26/Orthogroups.csv
   /share/pilot-johnsolk/killifish_RNAseq_fasta/Results_Nov26/Orthogroups.txt (OrthoMCL format)
   /share/pilot-johnsolk/killifish_RNAseq_fasta/Results_Nov26/Orthogroups_UnassignedGenes.csv

Gene trees:
   /share/pilot-johnsolk/killifish_RNAseq_fasta/Results_Nov26/Orthologues_Nov27/Gene_Trees/

Rooted species tree:
   /share/pilot-johnsolk/killifish_RNAseq_fasta/Results_Nov26/Orthologues_Nov27/SpeciesTree_rooted.txt

Species-by-species orthologues:
   /share/pilot-johnsolk/killifish_RNAseq_fasta/Results_Nov26/Orthologues_Nov27/Orthologues/

Orthogroup statistics:
   Statistics_PerSpecies.csv   Statistics_Overall.csv   Orthogroups_SpeciesOverlaps.csv

OrthoFinder assigned 2542304 genes (91.4% of total) to 133965 orthogroups. Fifty percent of all genes were in orthogroups with 43 or more genes (G50 was 43) and were contained in the largest 18142 orthogroups (O50 was 18142). There were 15603 orthogroups with all species present and 7 of these consisted entirely of single-copy genes.

CITATION:
 When publishing work that uses OrthoFinder please cite:
 Emms D.M. & Kelly S. (2015), Genome Biology 16:157

 If you use the species tree in your work then please also cite:
 Emms D.M. & Kelly S. (2017), MBE 34(12): 3267-3278
 Emms D.M. & Kelly S. (2018), bioRxiv https://doi.org/10.1101/267914
```
Next step, figure out what it all means!
