Title: Killifish PromethION data sharing
Date: 2018-10-08
Category: data
Tags: sequencing, killifish, Oxford Nanopore Technologies
Author: Lisa K. Johnson
Summary: Sharing PromethION sequencing data from 4 species of Fundulus killifish

We have new long-read [ONT PromethION](https://nanoporetech.com/products/promethion) genomic DNA sequencing data from 4 species of killifish! While we're working with it, generating new Illumina PE150 data for each species, and preparing a manuscript, we wanted to share this information and data in case it is useful for anyone. Since there are not many PromethION instruments out there, these are probably some of the first of its kind data being shared generated from organisms that are not human.

# Thank you!
This is a collaborative effort between my two advisors, Dr. Titus Brown and Dr. Andrew Whitehead. (Seriously, a grad student could not be luckier than me to have two great, supportive advisors and collaborative support from lab members and colleagues. I can't believe how lucky I am.) Other contributors are [Dr. Ruta Madhusudan](http://dnatech.genomecenter.ucdavis.edu/people/) and [Tony Gill](https://twitter.com/tonygill3?lang=en). Special thanks to [PoreCamp, Texas](http://porecamp.github.io/texas/), [Dr. Charlie Johnson](https://twitter.com/BioMath?lang=en) and [Dr. Richard Metz](https://twitter.com/DNA_hole?lang=en) at [Texas A&M Agrilife Research Sequencing Facility](http://www.txgen.tamu.edu/#aboutus) for Illumina NovaSeq data from [Fundulus olivaceus](http://txstate.fishesoftexas.org/fundulus%20olivaceus.htm). And [Dr. David Duvernell at Missouri University of Science & Technology](http://people.mst.edu/faculty/duvernelld/index.html) for sending us all of these fish to us, live! THANK YOU to support provided by the Moore Foundation Data-Driven-Discovery investigator award to Dr. Titus Brown.

# Questions for the community

The purpose of this blog post is mainly to share our plan and seek advice. Since this is a new type of data, we're in somewhat unchartered territory here about some things:

* What does one do with the millions of raw fast5 signal files from the PromethION? Our basecalled fastq are 70-100 GB, but the directory of raw fast5 signal files is >600 GB for each run! Wow. To our knowledge, the only software that uses these files to polish the reads is [Nanopolish](https://github.com/jts/nanopolish) by [Jared Simpson's group](https://twitter.com/jaredtsimpson?lang=en). Since the files are so large, I haven't tried using Nanopolish yet. It it a good idea to try? What is the best way to share >600 GB of millions of fast5 files with peopel who might be interested in using it?
* Data are unfiltered. Should we remove DNA CS lambda phage spike-in before posting? Remove other reads suspected as errors (>900 Mb)?
* Is sharing these fastq data useful for anyone? Would anyone be interested in collaborating on an [assemblathon](https://academic.oup.com/gigascience/article/2/1/2047-217X-2-10/2656129)-like project?
* Where is the best place to share these data? Even though [zenodo has recently upgraded their file size limits to 50 GB](http://help.zenodo.org/), it is not a good place to store these data based on their response to my request:
"""
Dear Lisa,

We're terribly sorry for the delay in response. Upload quota increase requests for bigger datasets have to be passed up the hierarchy, which usually takes time.

I'm sorry to inform you that we, unfortunately, will not be able to host the dataset as it surpasses our max allowed quota increase too much. By default, we provide a one-time quota increase up to 100GB for a dataset that will be cited from a peer-reviewed article.

Zenodo is a free-to-use service, an in order to keep it this way, we have to restrict the incoming data volume rate as very large datasets contribute significantly to the overall data volume in Zenodo. Unfortunately, at this point, we also cannot receive payment for quota increases, though we do hope that this will be possible in the future, at which point we will announce this possibility.
"""

**Any other advice, questions, or comments?**

# New PromethION instrument at the UC Davis Genome Center
In [May 2018](https://twitter.com/meganamsu/status/997216050247864322), the [UC Davis Genome Center](http://dnatech.genomecenter.ucdavis.edu/) received its PromethION from Oxford Nanopore! This was part of a collaborative effort to acquire this instrument through the [PromethION early release program (PEAP)](https://www.genomeweb.com/sequencing-technology/early-access-oxford-nanopore-promethion-requires-75k-deposit-24k-access-fee#.W70sSpNKhTY) back in early 2016 by Genome Center faculty: Dr. Richard Michelmore, Dr. Megan Dennis (who is on my dissertation committee!), and Dr. Titus Brown.

[I have been working with the ONT MinION](https://twitter.com/monsterbashseq/status/963592729241382912) to see if it is feasible to use for genome assemblies of eukaryotic organisms. Turns out that it is not. [Porecamp was very helpful for answering questions](https://twitter.com/monsterbashseq/status/959153636445073408) The PromethION has been great for our purposes. [MinION Costs](). Eukaryotic genomes are large. To achieve 30-50x coverage for a genome assembly of a 1-1.1 Gb killifish genome, we would need to use 30-50 MinION flowcells. At a [cost of $800-900/flowcell](https://docs.google.com/spreadsheets/d/1yBncz75kgwExCXy7sC9LsMaDGs8OJJJGg9f4o3DcoQE/edit?usp=sharing), that is not a reasonably affordable genome ($24,000-45,000). Whereas it took [39 flowcells to get ~30x (90Gb) coverage of the human genome](https://www.biorxiv.org/content/early/2017/04/20/128835), the PromethION has sequenced [~80x coverage of human genome](https://gigabaseorgigabyte.wordpress.com/2018/05/24/promethion-human-genome-na19240/). PromethION promises [higher yield](https://nanoporetech.com/about-us/news/promethion-wild-high-yield-data-announcements-human-and-plant-dna) than MinION, allowing it to be more feasible for larger genomes >1 Gb. The cost of one [PromethION run at UC Davis](http://dnatech.genomecenter.ucdavis.edu/nanopore-sequencing-ont-promethion/) is $2,880.

And the avalance of data commences!

# Four species of *Fundulus* killifish

<img style="float: right;" src="https://images-na.ssl-images-amazon.com/images/I/81SpSNXaFfL.jpg" height="200">

These species of killifish have never been sequenced before. (We have a genome of the [classic physiological model](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2128618/) species, [*Fundulus heteroclitus*](http://uswest.ensembl.org/Fundulus_heteroclitus/Info/Index). But we do not have any genomic information from other important freshwater *Fundulus* species. These are ) Live samples were gratiously shipped to UC Davis by David Duvernell.

*Fundulus heteroclitus* killifish is well-known for its extreme [resilience](https://onlinelibrary.wiley.com/doi/epdf/10.1111/gcb.14386?author_access_token=36pNDS0SUqB6m5gUMREijota6bR2k8jH0KrdpFOxC66a2-XVORYljQz2FrLUSYMKsfdfVt3Mxuv_BJEgeCfbetZEseZqrz14rbQRWL4-Ge1jjHI83VoyEXRrMQWXA1A3) and tolerant [model](https://www.sciencedirect.com/science/article/pii/S1744117X0700072X), capable of [rapid adaptation](http://science.sciencemag.org/content/354/6317/1305), long studied for physiological resilience to salinity change.

* [Whitehead et al. 2010. Phylogeny of killifish osmotic tolerance](https://www.ncbi.nlm.nih.gov/pubmed/20100216)
* [Whitehead et al. 2011. Genomic mechanisms of evolved physiological plasticity in killifish distributed along an environmental salinity gradient](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3076822/)
* [Brennan et al. 2015. Reciprocal osmotic challenges reveal mechanisms of divergence in phenotypic plasticity in the killifish Fundulus heteroclitus](http://jeb.biologists.org/content/218/8/1212)
* Recently, the paper [Rabosky et al. 2018. An inverse latitudinal gradient in speciation rate for marine fishes.](https://www.nature.com/articles/s41586-018-0273-1) used data from the [Fundulidae tree of life](https://fishtreeoflife.org/taxonomy/family/Fundulidae/)

# DNA extractions

* [Ultra-long read sequencing protocol by Josh Quick and Nick Loman](https://www.protocols.io/view/ultra-long-read-sequencing-protocol-for-rad004-mrvc566?version_warning=no) 
* [Agilent #200600 DNA extraction kit](https://www.agilent.com/cs/library/usermanuals/public/200600.pdf) found by Tony
* Tony and I found that fresh is better than frozen
* Tried [Qiagen](https://www.qiagen.com/us/resources/resourcedetail?id=a9e6a609-4600-4b03-afbd-974318590ce5&lang=en) - p. 39 "DNA purification from tissue using the Gentra puregene Tissue Kit", 
* Also tried [SDS/Phenol:Chloroform](https://www.protocols.io/view/ultra-long-read-sequencing-protocol-for-rad004-mrxc57n)
* Ruta found that adding urea to lysis buffer helped [Wasko et al. 2003](https://www.ncbi.nlm.nih.gov/pubmed/14641478)
* DNA was difficult to get into solution. (show pic)
* With the MinION, our sample was not utilizing the pores as efficiently as it could be. Only getting ~1 Gb out of a single flowcell. Others have gotten ~10 Gb. Something weird with our samples. ONT has mentioned that this is a fish-specific sample issue. Has anyone else found this to be the case with fish hmw DNA? Also brittle and breaking easily.

(show gel of smeary lanes)

[Good pf gel, thanks to Tony Gil](https://twitter.com/monsterbashseq/status/852655158160904192)

[Porecamp sequencing animals](https://twitter.com/nanopore/status/872910510508953605)

# Sequencing

ONT data:

Species |  bases called |  n reads | avg length | largest | N50 |
--- | --- | --- | --- | --- | --- | --- | --- |
Adinia xenica | 38,467,326,719 | 15,704,522 | 2,449 | 953,774 | 5,733, n = 1,373,426 |
Fundulus nottii | 33,440,866,723 | 5,160,367 | 6,480 | 667,947 | 12,995, n=700,534 |
F. catenatus | 40,274,806,587 | 23,701,206 | 1,699 | 590,485 | 3,439, n = 2,687,295 |


Overall, here is the status of this project:

Species |  native physiology | clade | ONT data |  Illumina | best N50 | |
--- | --- | --- | --- | --- | --- | --- | --- | 
*Fundulus catenatus* | Freshwater | 1 | TBD | TBD | TBD | <img src="https://nature.mdc.mo.gov/sites/default/files/styles/centered_full/public/media/images/2010/05/northern_studfish.jpg?itok=kLYZJ6V4">
*Adinia xenica* | Marine  | 3 | 38 Gb | TBD | 369,038; n = 794 | <img src="https://biodiversity.utexas.edu/images/easyblog_articles/3681/adinia_xenica.jpg"> 
*Fundulus olivaceus* | Freshwater | 3 | 4.9 Gb | 180 Gb | 42,823; n=7,616 (Masurca) | <img src="https://nature.mdc.mo.gov/sites/default/files/styles/centered_full/public/media/images/2010/05/blackspotted_topminnow.jpg?itok=1V78GVZI">
*Fundulus nottii* | Freshwater | 3 | 33 Gb | TBD | 2 Mb; n = 151 | [<img src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcT1B0oYaT8isujaT2qLP3decYTRvuiLZGvinMEEkR0algWfnrzV">](http://www.nanfa.org/ac/fundulus-notti-starhead-toopminnow.pdf)

Other species of freshwater and brackish water killifish (*Lucania parva* and *Lucania goodei*) from Clade 2 are being sequenced by [Dr. Becky Fuller's lab at U. Illinois Urbana-Champaign](http://beckyfullerlab.weebly.com/).

# Preliminary data products

Rough assemblies and QC:

https://osf.io/zjv86/

# Our plan from here:

## 1. Post data

* [European Nucleotide Archive (ENA)-EBI](https://www.ebi.ac.uk/ena/submit/sequence-submission) 
* SRA-NCBI - is ridiculous

## 2. Data note

Either:
- [F1000 research](https://f1000research.com/for-authors/article-guidelines/data-notes)
- [GigaScience](https://academic.oup.com/gigascience/pages/data_note)

## 3. Do the science!
Comparative genomics. Scan for regulatory regions of interest, e.g. [salinity-responsive enhancing elements (OSRE)](http://www.pnas.org/content/114/13/E2729)


# References

[A Fast Approximate Algorithm for Mapping Long Reads to Large Reference Databases](https://www.biorxiv.org/content/early/2017/03/24/103812)

>The anticipated increasing speed of nanopore sequencing ('fast mode') and
the scaling up of the MinION to 3,000 channels, and the PromethION with 144,000 channels, will challenge the implementation of 'Read Until' in real time and require algorithmic enhancements and computational power

[Real time selective sequencing using nanopore technology](https://www.biorxiv.org/content/early/2016/02/03/038760)

[Nanopore sequencing and assembly of a human genome with ultra-long reads](https://www.biorxiv.org/content/biorxiv/early/2017/04/20/128835.full.pdf)

[Nice slides (Benjamin Schwessinger), explainig wet lab comparisons for hmw DNA prep](https://f1000research.com/slides/7-367)

[Finding Nemo: hybrid assembly with Oxford Nanopore and Illumina reads greatly improves the clownfish (Amphiprion ocellaris) genome assembly](https://academic.oup.com/gigascience/article/7/3/gix137/4803946)

[Nematode genomes, ONT - From Erich Schwartz](http://www.g3journal.org/content/early/2018/08/07/g3.118.200160)

Use spades? https://www.biostars.org/p/233062/

Papers (recommended from [Biostars](https://www.biostars.org/p/233062/))

* https://www.biorxiv.org/content/early/2016/05/22/054783
* https://gigascience.biomedcentral.com/articles/10.1186/s13742-015-0076-3
* https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0951-y

[Tyler et al. 2018. Evaluation of Oxford Nanopore’s MinION Sequencing Device for Microbial Whole Genome Sequencing Applications](https://www.nature.com/articles/s41598-018-29334-5.pdf)

[Accurate characterization of expanded tandem repeat length and sequence through whole genome long-read sequencing on PromethION.](https://www.biorxiv.org/content/early/2018/10/09/439026)

# List of available Tools:

With varying degrees of difficulty installing and using:

### QC & Evaluation:
* [Porechop](https://github.com/rrwick/Porechop): removing adapters
* [NanoPlot](https://anaconda.org/bioconda/nanoplot)
* [sourmash](http://sourmash.readthedocs.io/en/latest/tutorials.html) to classify and compare contig sequence similarity between assemblies, compare reads to assemblies. IDEA FROM TITUS (12/5/2017): use sourmash to cluster contigs by similarity, then do separate dotplots:
https://github.com/ctb/2017-sourmash-cluster/blob/master/cluster-big-hashsets.ipynb
* [BUSCO](chrome-extension://oemmndcbldboiebfnladdacbdfmadadm/https://gitlab.com/ezlab/busco/raw/master/BUSCO_v3_userguide.pdf) content evaluation
* [Nanocomp](https://github.com/wdecoster/nanocomp), compare multiple ONT runs
* [mashmap](https://github.com/marbl/MashMap), for visualization (replaces nucmer)

### Assemblers:

* [Canu v1.6](https://github.com/marbl/canu/releases): [Quick start](http://canu.readthedocs.io/en/latest/quick-start.html), [tutorial](https://github.com/marbl/canu/blob/master/documentation/source/tutorial.rst) and [paper](http://genome.cshlp.org/content/27/5/722.full)
* Masurca: [new version 3.2.2: (ftp://ftp.genome.umd.edu/pub/MaSuRCA/latest/) (http://www.genome.umd.edu/masurca.html) ([supports ONT reads](http://masurca.blogspot.com.au/2017/05/please-upgrade-to-masurca-release-322.html) but...no instructions for ONT yet) here is a [presentation](https://www.youtube.com/watch?v=BEsDkjYKHCE&feature=youtu.be) and [tweet](https://twitter.com/conchoecia/status/920771453728854016) indicating possible to use ONT reads...), [quick start](http://bit.ly/2zl7G5Q)
* [SPAdes hybrid assembly](http://cab.spbu.ru/files/release3.11.1/manual.html) (more for bacteria, according to Lex Nederbragt)
* [Alpaca, hybrid assembly](https://github.com/VicugnaPacos/ALPACA/) and [paper](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-017-3927-8)

### Polishing, scaffolding and consensus improvement:
* [Unicycler polish](https://github.com/rrwick/Unicycler/blob/master/docs/unicycler-polish.md) ([paper])(http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005595)
* [Nanopore polish, input ONT assembly with Illumina data](https://github.com/nanoporetech/ont-assembly-polish)
* [SSPACE-LongRead scaffolding](https://www.baseclear.com/genomics/bioinformatics/basetools/SSPACE-longread)
(no instructions, but a [paper](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-15-211)!), here are [my instructions for SSPACE-LongRead](https://github.com/ljcohen/hybrid_genome_assembly/blob/master/SSPACE/README.md)
* [Nanopolish](https://github.com/jts/nanopolish)
* [Nanocorr](https://github.com/jgurtowski/nanocorr): error correction (See [presentation](http://schatzlab.cshl.edu/presentations/2015/2015.02.28.AGBT%20Nanocorr%20Assembly.pdf))
* [racon](https://github.com/isovic/racon)
* [pilon](https://github.com/broadinstitute/pilon)


### Misc

* [npScarf](https://www.nature.com/articles/ncomms14515)
* [pipeline from Drosophila](https://github.com/danrdanny/Drosophila15GenomesProject/blob/fe669468b4b37580daa213e8bb484281580e5b79/script_runPilon.pl)

# Rough Draft of Workflow:

Create conda environment:
```
conda create -n ONT python=3
source activate ONT
```

[Minimap2](https://github.com/lh3/minimap2) (minimap2-2.12) and [miniasm](https://github.com/lh3/miniasm)
```
conda install assembly-stats minimap2 miniasm
```
[Miniasm](https://github.com/lh3/miniasm)
```
source ~/.bashrc
source activate ONT
cd /home/ljcohen/A_xenica_promethION/
# Overlap
minimap2 -x ava-ont -t8 A_xenica_combined.fastq | gzip -1 > A_xenica_reads.paf.gz
# Layout
miniasm -f A_xenica_combined.fastq A_xenica_reads.paf.gz > reads.gfa
```

Versions:
```
(ONT) ljcohen@c8-67:~/A_xenica_promethION$ miniasm -V
0.2-r128
(ONT) ljcohen@c8-67:~/A_xenica_promethION$ minimap2 -V
2.12-r827
```

pilon to improve consensus:
```
module load bwa samtools pilon racon
bwa index <genome.fasta>
bwa mem -t 4 genome.fasta R1.fq R2.fq | samtools view -Sb - | samtools sort - -o ecto_old_paired_qc.sorted.bam
samtools index ecto_old_paired_qc.sorted.bam
pilon --genome F_olivaceus.canu.ONT.unitigs.fasta --bam F_olivaceus_ONT_polished.sorted.bam --threads 8
```

BUSCO for evaluation:
```
module load busco
cp -r /software/augustus/3.3.0/lssc0-linux/config.copy busco/
export AUGUSTUS_CONFIG_PATH=/share/pilot-johnsolk/busco/config.copy
-l${BUSCO_LINEAGE}/
run_BUSCO.py -i F_olivaceus_megahit_Ilumina_final.contigs.fa -l ${BUSCO_LINEAGE}/eukaryota_odb9 -o F_olivaceus_megahit --mode genome -c 8
```

Visualize, run [mashmap](https://github.com/marbl/MashMap):
```
mashmap \
-r /mnt/home/ljcohen/reference/GCF_000826765.1_Fundulus_heteroclitus-3.0.2_genomic.fna \
-q ../../masurca_Folivaceus/CA.mr.41.15.15.0.02/final.genome.scf.fasta
```
Plot:
```
generateDotPlot png large mashmap.out
```
But will have to edit the `out.gp` file, because of reasons.
```
vi out.gp
```
1. Delete word 'tiny' in first line
2. 3 lines with word 'mouse' in them.

Then run:
```
gnuplot out.gp
```
Which will produce the png file.
