Title: Killifish PromethION data sharing
Date: 2018-10-08
Category: Python
Tags: sequencing, killifish, Oxford Nanopore Technologies
Author: Lisa K. Johnson
Summary: Sharing PromethION sequencing data from 4 species of Fundulus killifish

We have some new long-read [ONT PromethION](https://nanoporetech.com/products/promethion) genomic DNA sequencing data from killifish! While we're working with the data, generating Illumina PE150 data for each, and preparing a manuscript, we wanted to share this information and data in case it is useful for anyone. Since there are not many PromethION instruments out there, these are probably some of the first of its kind data being shared generated from organisms that are not human.

# Thank you!
This is a collaborative effort between my two advisors, Dr. Titus Brown and Dr. Andrew Whitehead. (Seriously, a grad student could not be luckier than me to have two great, supportive advisors and collaborative support from lab members and colleagues. I can't believe how lucky I am.) Other contributors are [Dr. Ruta Madhusudan](http://dnatech.genomecenter.ucdavis.edu/people/) and [Tony Gill](https://twitter.com/tonygill3?lang=en). Special thanks to [PoreCamp, Texas](http://porecamp.github.io/texas/), [Charlie Johnson](https://twitter.com/BioMath?lang=en) and [Richard Metz](https://twitter.com/DNA_hole?lang=en) at [Texas A&M Agrilife Research Sequencing Facility](http://www.txgen.tamu.edu/#aboutus) for Illumina NovaSeq data from [Fundulus olivaceus](http://txstate.fishesoftexas.org/fundulus%20olivaceus.htm). And [Dr. David Duvernell at Missouri University of Science & Technology](http://people.mst.edu/faculty/duvernelld/index.html) for sending us all of these fish to us, live! THANK YOU to support provided by the Moore Foundation Data-Driven-Discovery investigator award to Dr. Titus Brown.

# New PromethION instrument at the UC Davis Genome Center
In [May 2018](https://twitter.com/meganamsu/status/997216050247864322), the [UC Davis Genome Center](http://dnatech.genomecenter.ucdavis.edu/nanopore-sequencing-ont-promethion/) received its PromethION from Oxford Nanopore! This was part of a collaborative effort to acquire this instrument through the [PromethION early release program (PEAP)](https://www.genomeweb.com/sequencing-technology/early-access-oxford-nanopore-promethion-requires-75k-deposit-24k-access-fee#.W70sSpNKhTY) back in early 2016 by Genome Center faculty: Dr. Richard Michelmore, Dr. Megan Dennis (who is on my dissertation committee!), and Dr. Titus Brown.

I have been working with the ONT MinION to see if it is feasible to use for genome assemblies of eukaryotic organisms. 

https://twitter.com/monsterbashseq/status/963592729241382912

* MinION Costs: https://twitter.com/monsterbashseq/status/963154668942589952
* I went to Porecamp: https://twitter.com/monsterbashseq/status/959153636445073408
* Turns out that it is not. But the PromethION has been great for our purposes.
* PromethION promises [higher yield](https://nanoporetech.com/about-us/news/promethion-wild-high-yield-data-announcements-human-and-plant-dna) than MinION, allowing it to be more feasible for 

# *Fundulus* killifish

<img align="right" height="200" src="https://images-na.ssl-images-amazon.com/images/I/81SpSNXaFfL.jpg">

These species of killifish have never been sequenced before. Live samples were gratiously shipped to UC Davis by 

https://twitter.com/monsterbashseq/status/1048308106596245504

Very important fish for comparative genomics:

* [Whitehead et al. 2010. Phylogeny of killifish osmotic tolerance](https://www.ncbi.nlm.nih.gov/pubmed/20100216)
* [Whitehead et al. 2011. Genomic mechanisms of evolved physiological plasticity in killifish distributed along an environmental salinity gradient](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3076822/)
* [Brennan et al. 2015. Reciprocal osmotic challenges reveal mechanisms of divergence in phenotypic plasticity in the killifish Fundulus heteroclitus](http://jeb.biologists.org/content/218/8/1212)
* Recently, [Fundulidae tree of life](https://fishtreeoflife.org/taxonomy/family/Fundulidae/)

# DNA extractions

* long-reads protocol by Joshua Quick and Nick Loman: https://twitter.com/Scalene/status/955091398285582337
* Agilent protocol, found by Tony
* Fresh is better than frozen
* Tried other protocols
* Ruta found that adding urea to lysis buffer helped
* DNA was difficult to get into solution.
* With the MinION, our sample was not utilizing the pores as efficiently as it could be. ONT has mentioned that this is a fish-specific sample issue. Has anyone else found this to be the case with fish hmw DNA? Brittle and breaking easily. 
* Show gels.

https://twitter.com/monsterbashseq/status/852655158160904192

https://twitter.com/nanopore/status/872910510508953605

# Sequencing

Species |  native physiology | clade | ONT data |  Illumina | best N50 | |
--- | --- | --- | --- | --- | --- | --- | --- | 
*Fundulus catenatus* | Freshwater | 1 | TBD | TBD | TBD | <img src="https://nature.mdc.mo.gov/sites/default/files/styles/centered_full/public/media/images/2010/05/northern_studfish.jpg?itok=kLYZJ6V4">
*Adinia xenica* | Marine  | 3 | 38 Gb | TBD | 369,038; n = 794 | <img src="https://biodiversity.utexas.edu/images/easyblog_articles/3681/adinia_xenica.jpg"> 
*Fundulus olivaceus* | Freshwater | 3 | 4.9 Gb | 180 Gb | 42,823; n=7,616 (Masurca) | <img src="https://nature.mdc.mo.gov/sites/default/files/styles/centered_full/public/media/images/2010/05/blackspotted_topminnow.jpg?itok=1V78GVZI">
*Fundulus nottii* | Freshwater | 3 | 33 Gb | TBD | 2 Mb; n = 151 | [<img src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcT1B0oYaT8isujaT2qLP3decYTRvuiLZGvinMEEkR0algWfnrzV">](http://www.nanfa.org/ac/fundulus-notti-starhead-toopminnow.pdf)

Other species of freshwater and brackish water killifish (*Lucania parva* and *Lucania goodei*) from Clade 2 are being sequenced by [Dr. Becky Fuller's lab at U. Illinois Urbana-Champaign](http://beckyfullerlab.weebly.com/).

Fundulus catenatus (freshwater native, Clade 1) read stats:
```
(ONT) johnsolk@barbera:/share/pilot-johnsolk/ONT_killifish/F_catenatus$ assembly-stats F_cat.fq 
stats for F_cat.fq
sum = 40274806587, n = 23701206, ave = 1699.27, largest = 590485
N50 = 3439, n = 2687295
N60 = 2464, n = 4076415
N70 = 1739, n = 6027668
N80 = 1178, n = 8844065
N90 = 718, n = 13205064
N100 = 5, n = 23701206
N_count = 0
Gaps = 0
```

Fundulus nottii (freshwater antive, Clade 3) read stats:
```
(ONT) [ljcohen@dev-intel14 F_nottii_PromethION]$ assembly-stats Fnottii_combined.fastq
stats for Fnottii_combined.fastq
sum = 33440866723, n = 5160367, ave = 6480.33, largest = 667947
N50 = 12995, n = 700534
N60 = 9878, n = 996409
N70 = 7391, n = 1388376
N80 = 5305, n = 1921978
N90 = 3399, n = 2702711
N100 = 5, n = 5160367
N_count = 0
Gaps = 0
```


Adinia xenica (marine native, Clade 3) read stats:
```
(ONT) [ljcohen@dev-intel14 A_xenica_PromethION]$ assembly-stats A_xenica_combined.fastq
stats for A_xenica_combined.fastq
sum = 38467326719, n = 15704522, ave = 2449.44, largest = 953774
N50 = 5733, n = 1373426
N60 = 3884, n = 2193610
N70 = 2601, n = 3408866
N80 = 1661, n = 5265095
N90 = 952, n = 8322903
N100 = 5, n = 15704522
N_count = 0
Gaps = 0
```

## Preliminary data products

Rough assemblies and QC:

https://osf.io/zjv86/

# Our plan from here:

## 1. Post data

* [European Nucleotide Archive (ENA)-EBI](https://www.ebi.ac.uk/ena/submit/sequence-submission) 
* SRA-NCBI - is ridiculous

## 2. Data note

Either:
- [F1000 research](https://f1000research.com/for-authors/article-guidelines/data-notes)
- [GigaScience](https://academic.oup.com/gigascience/pages/data_note)

## 3. Do the science!
Comparative genomics. Scan for regulatory regions of interest, e.g. [salinity-responsive enhancing elements (OSRE)](http://www.pnas.org/content/114/13/E2729)

# Questions

* What to do with millions of fast5 files? Our fastq are 70-100 GB, but the directory of raw fast5 signal files are >600 GB. [Nanopolish](https://github.com/jts/nanopolish) uses the raw fast5 signal files to correct. How can these be shared efficiently?
* Data are unfiltered, should we remove DNA CS lambda phage spike-in?
* Where is the best place to share these data? Even though [zenodo has recently upgraded their file size limits to 50 GB](http://help.zenodo.org/), it is not a good place:
```
Dear Lisa,

We're terribly sorry for the delay in response. Upload quota increase requests for bigger datasets have to be passed up the hierarchy, which usually takes time.

I'm sorry to inform you that we, unfortunately, will not be able to host the dataset as it surpasses our max allowed quota increase too much. By default, we provide a one-time quota increase up to 100GB for a dataset that will be cited from a peer-reviewed article.

Zenodo is a free-to-use service, an in order to keep it this way, we have to restrict the incoming data volume rate as very large datasets contribute significantly to the overall data volume in Zenodo. Unfortunately, at this point, we also cannot receive payment for quota increases, though we do hope that this will be possible in the future, at which point we will announce this possibility.
```

* Is sharing the data useful for anyone? 
* Any advice, questions, or comments?
